---
title: "T7 warmx 16S Analysis 2021"
author: "Moriah Young"
date: "2023-01-31"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
rm(list = ls()) # clear working environment

# load packages
library(compositions)
library(cowplot)
library(cluster)
library(data.table)
library(devtools)
library(plyr)
library(dplyr)
library(doParallel)
library(factoextra)
library(ggplot2)
library(ggsignif)
library(gridExtra)
library(hillR)
library(microbiome)
library(multcomp)
library(nlme)
library(phyloseq)
library(phylosmith)
library(qiime2R)
library(tidyverse)
library(vegan)
library(zCompositions)
```

# Read QZA files into dataframe, re-format taxonomic tables, and re-upload them as .csv files
```{r}
SVs16S <- read_qza("/Users/moriahyoung/Downloads/16S-2021-merged-dada2table.qza")
SVs16S_merged_table <- SVs16S$data
write.csv(SVs16S_merged_table, file = "/Users/moriahyoung/Downloads/16S-2021-merged-dada2table.csv")
taxonomy16S <- read_qza("/Users/moriahyoung/Downloads/16S-taxonomy.qza")
tax16S<-taxonomy16S$data %>% as_tibble() %>%
  separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species")) %>%
  mutate(Phylum = replace_na(Phylum,"empty"))
write.csv(tax16S, file = "/Users/moriahyoung/Downloads/16S-taxonomy.csv", row.names = T)

# Write unfiltered data into a phyloseq object
data_16S_unfiltered <- read_csv2phyloseq(otu.file = "/Users/moriahyoung/Downloads/16S-2021-merged-dada2table.csv",
                                         taxonomy.file = "/Users/moriahyoung/Downloads/16S-taxonomy.csv",
                                         metadata.file = "/Users/moriahyoung/Desktop/bacterial_2021_metadata.csv")

summarize_phyloseq(data_16S_unfiltered)

# filter out non bacteria
data_16S_uf1 <- subset_taxa(data_16S_unfiltered, Kingdom == "Bacteria" | Kingdom == "Archaea")
data_16S_uf2 <- subset_taxa(data_16S_uf1, Kingdom != "Eukaryota")
data_16S_uf3 <- subset_taxa(data_16S_uf2, Order != "Chloroplast")
data_16S_uf4 <- subset_taxa(data_16S_uf3, Family != "Mitochondria")

summarize_phyloseq(data_16S_uf4)

```
For some reason code below works but above chunk does not
Read QZA files into dataframe, re-format taxonomic tables, and re-upload them as .csv files
```{r}
SVs16S <- read_qza("/Users/moriahyoung/Downloads/16S-2021-merged-dada2table.qza")
SVs16Stable <- SVs16S$data
write.csv(SVs16Stable, file = "/Users/moriahyoung/Downloads/16S-merged-table.csv")
taxonomy16S <- read_qza("/Users/moriahyoung/Downloads/16S-taxonomy.qza")
tax16S<-taxonomy16S$data %>% as_tibble() %>%
  mutate(Taxon=gsub("[a-z]__", "", Taxon)) %>% 
  separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species"))%>%
  mutate(Phylum=replace_na(Phylum,"empty"))
write.csv(tax16S, file = "/Users/moriahyoung/Downloads/16S-taxonomy.csv", row.names =F)

# create phyloseq object
data_16S_unfiltered <- read_csv2phyloseq(otu.file = "/Users/moriahyoung/Downloads/16S-merged-table.csv",
                                         taxonomy.file = "/Users/moriahyoung/Downloads/16S-taxonomy.csv",
                                         metadata.file = "/Users/moriahyoung/Desktop/bacterial_2021_metadata.csv")
summarize_phyloseq(data_16S_unfiltered)

# filter out non bacteria
data_16S_uf1 <- subset_taxa(data_16S_unfiltered, Kingdom == "Bacteria" | Kingdom == "Archaea")
data_16S_uf2 <- subset_taxa(data_16S_uf1, Kingdom != "Eukaryota")
data_16S_uf3 <- subset_taxa(data_16S_uf2, Order != "Chloroplast")
data_16S_uf4 <- subset_taxa(data_16S_uf3, Family != "Mitochondria")

summarize_phyloseq(data_16S_uf4)
```

```{r}
# Remove samples with extremely low read depth 
data_16S_uf5 <- prune_samples(sample_sums(data_16S_uf4)>=1000, data_16S_uf4)

# Export filtered data
write.csv(data_16S_uf5@otu_table, "16S-merged-table-filtered.csv")
write.csv(data_16S_uf5@tax_table, "16S-taxonomy.csv")
```

# Rarefaction
```{r}
# Prune SVs that are not present 5 times in at least 2 samples

data_16S_counts <- data_16S_filtered
data_16S_counts <- prune_samples(sample_sums(data_16S_counts)>=1000, data_16S_counts)
data_16S_filtered <- transform_sample_counts(data_16S_filtered, function(x) x/sum(x))
data_16S_counts <- filter_taxa(data_16S_counts, function(x) sum(x >5) > (0.01058201*length(x)), TRUE)
summarize_phyloseq(data_16S_counts)

sum(colSums(otu_table(data_16S_counts)))
sort(colSums(otu_table(data_16S_counts)))
```

