---
title: "T7 warmx 16S Analysis 2021"
author: "Moriah Young"
date: "2023-01-31"
output: pdf_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, echo=FALSE, message=FALSE}
rm(list = ls()) # clear working environment

# load packages
library(compositions)
library(cowplot)
library(cluster)
library(data.table)
library(devtools)
library(plyr)
library(dplyr)
library(doParallel)
library(factoextra)
library(ggplot2)
library(ggsignif)
library(gridExtra)
library(hillR)
library(microbiome)
library(multcomp)
library(nlme)
library(phyloseq)
library(phylosmith) ##
library(qiime2R)
library(tidyverse)
library(vegan)
library(zCompositions)
library(microViz) ##
library(DESeq2) ##
library(ANCOMBC) ##
library(permute)
library(pheatmap)
library(plotrix)

# Set working directory from .Renviron
dir <- Sys.getenv("DATA_DIR")
list.files(dir)
```

Read QZA files into dataframe, re-format taxonomic tables, and re-upload them as .csv files
```{r}
# Code below you only need to do once so it's # out as to not have to run every time
#SVs16S <- read_qza("/Users/moriahyoung/Downloads/16S-2021-merged-dada2table.qza")
#SVs16Stable <- SVs16S$data
#write.csv(SVs16Stable, file = "/Users/moriahyoung/Downloads/16S-merged-dada2table.csv")

#taxonomy16S <- read_qza("/Users/moriahyoung/Downloads/16S-taxonomy.qza")
#tax16S <- taxonomy16S$data %>% as_tibble() %>%
#  mutate(Taxon=gsub("[a-z]__", "", Taxon)) %>% 
#  separate(Taxon, sep=";", c("Kingdom","Phylum","Class","Order","Family","Genus","Species"))%>%
#  mutate(Phylum=replace_na(Phylum,"empty"))
#write.csv(tax16S, file = "/Users/moriahyoung/Downloads/16S-taxonomy.csv", row.names =F)

physeq_tree <- qza_to_phyloseq(
        tree = "/Users/moriahyoung/Desktop/physeq_object/16S-2021-fasttree-rooted-tree.qza"
        )
physeq_tree

# create phyloseq object
#data_16S_unfiltered <- read_csv2phyloseq(otu.file = "/Users/moriahyoung/Downloads/16S-merged-dada2table.csv",
#                                         taxonomy.file = "/Users/moriahyoung/Downloads/16S-taxonomy.csv",
#                                         metadata.file = "/Users/moriahyoung/Desktop/16S_2021_metadata.csv")
#summarize_phyloseq(data_16S_unfiltered)
```

```{r}
# filter data

# filter out non bacteria
#data_16S_uf1 <- subset_taxa(data_16S_unfiltered, Kingdom == "Bacteria" | Kingdom == "Archaea")
#data_16S_uf2 <- subset_taxa(data_16S_uf1, Kingdom != "Eukaryota")
#data_16S_uf3 <- subset_taxa(data_16S_uf2, Order != "Chloroplast")
#data_16S_uf4 <- subset_taxa(data_16S_uf3, Family != "Mitochondria")

#summarize_phyloseq(data_16S_uf4)

# Remove samples with extremely low read depth 
# data_16S_uf5 <- prune_samples(sample_sums(data_16S_uf4)>=1000, data_16S_uf4)

# Export filtered data
# write.csv(data_16S_uf5@otu_table, "/Users/moriahyoung/Desktop/16S-merged-table-filtered.csv")
# write.csv(data_16S_uf5@tax_table, "/Users/moriahyoung/Desktop/16S-taxonomy-filtered.csv")
# upload these to REX google shared drive

taxonomy <- read.csv("/Users/moriahyoung/Desktop/physeq_object/16S-taxonomy-filtered.csv")
otu <- read.csv("/Users/moriahyoung/Desktop/physeq_object/16S-merged-table-filtered.csv")

# Create phyloseq objects of filtered data (16S)
data_16S_filtered <- read_csv2phyloseq(otu.file = "/Users/moriahyoung/Desktop/physeq_object/16S-merged-table-filtered.csv",
                                       taxonomy.file = "/Users/moriahyoung/Desktop/physeq_object/16S-taxonomy-filtered.csv",
                                       metadata.file = "/Users/moriahyoung/Desktop/physeq_object/16S_2021_metadata.csv")
summarize_phyloseq(data_16S_filtered)

# merge the phyloseq object with the phylogenetic tree with the other phyloseq object
data_16S_filtered <- merge_phyloseq(data_16S_filtered, physeq_tree)
summarize_phyloseq(data_16S_filtered)

# order sample data - "Drought" and "Subplot_Descriptions"
sample_data(data_16S_filtered)$Drought <- ordered(sample_data(data_16S_filtered)$Drought, c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"))
sample_data(data_16S_filtered)$Subplot_Descriptions <- ordered(sample_data(data_16S_filtered)$Subplot_Descriptions, c("irrigated_control", "ambient", "warmed", "warmed_insecticide", "insecticide", "drought", "warmed_drought", "drought_insecticide", "warmed_drought_insecticide"))
```

# Rarefaction Curves
```{r}
data_16S_counts <- data_16S_filtered
# remove samples that have less than 1000 reads
data_16S_counts <- prune_samples(sample_sums(data_16S_counts)>=1000, data_16S_counts) # this is already done above...
#data_16S_filtered <- transform_sample_counts(data_16S_filtered, function(x) x/sum(x))
# Prune SVs that are not present 5 times in at least 2 samples --> remove taxa not seen more than 5 times in at least ~1% of the samples. This protects against an OTU with small mean & trivially large C.V.
#data_16S_counts <- filter_taxa(data_16S_counts, function(x) sum(x >5) > (0.01058201*length(x)), TRUE)
summarize_phyloseq(data_16S_counts)

sum(colSums(otu_table(data_16S_counts)))
sort(colSums(otu_table(data_16S_counts)))
# min = 2488
# max = 202,979

# https://github.com/joey711/phyloseq/issues/143
# Rarefaction Curve Function
calculate_rarefaction_curves <- function(psdata, measures, depths, parallel=T) {
        require('plyr') # ldply
        require('reshape2') # melt
        require('doParallel')
  
# set parallel options if required
if (parallel) {
        paropts  <- list(.packages=c("phyloseq", "reshape2"))
        } else {
                paropts  <- NULL
                }
  
estimate_rarified_richness <- function(psdata, measures, depth) {
    if(max(sample_sums(psdata)) < depth) return()
    psdata <- prune_samples(sample_sums(psdata) >= depth, psdata)
    
    rarified_psdata <- rarefy_even_depth(psdata, depth, verbose = FALSE)
    
    alpha_diversity <- estimate_richness(rarified_psdata, measures = measures)
    
    # as.matrix forces the use of melt.array, which includes the Sample names (rownames)
    molten_alpha_diversity <- melt(as.matrix(alpha_diversity), varnames = c('Sample', 'Measure'), value.name = 'Alpha_diversity')
    
    molten_alpha_diversity
  }
  
  names(depths) <- depths # this enables automatic addition of the Depth to the output by ldply
  rarefaction_curve_data <- ldply(depths, estimate_rarified_richness, psdata = psdata, measures = measures, .id = 'Depth', .progress = ifelse(interactive() && ! parallel, 'text', 'none'), .parallel=parallel, .paropts=paropts)
  
  # convert Depth from factor to numeric
  rarefaction_curve_data$Depth <- as.numeric(levels(rarefaction_curve_data$Depth))[rarefaction_curve_data$Depth]
  
  rarefaction_curve_data
}

# Summarize alpha diversity
rarefaction_curve_data <- calculate_rarefaction_curves(data_16S_counts, c('Observed', 'Shannon'), 
                                                       rep(c(1, 10, 100, 1000, 1:100 * 10000), each = 10))

rarefaction_curve_data_summary <- ddply(rarefaction_curve_data, c('Depth', 'Sample', 'Measure'), summarise, Alpha_diversity_mean = mean(Alpha_diversity), Alpha_diversity_sd = sd(Alpha_diversity))

# Add sample data
rarefaction_curve_data_summary_verbose <-  merge(rarefaction_curve_data_summary %>% mutate(Sample = gsub("[.]", "-", Sample)),
                                                 data.frame(sample_data(data_16S_counts)) %>% 
                                                         rownames_to_column(var = "rowname"),
                                                 by.x = 'Sample', by.y = 'rowname')

discrete_palettes <- list(RColorBrewer::brewer.pal(3, "Set2"))

rarefaction_curve_data_summary_verbose$Subplot_Descriptions <- factor(rarefaction_curve_data_summary_verbose$Subplot_Descriptions)

# plot
curve_16S_facet <- 
        ggplot(rarefaction_curve_data_summary_verbose %>% filter(Measure == "Observed"), aes(x = Depth, y = Alpha_diversity_mean,  ymin = Alpha_diversity_mean - Alpha_diversity_sd, ymax = Alpha_diversity_mean + Alpha_diversity_sd, colour = Subplot_Descriptions, group = Sample)) + 
        geom_line(alpha=1) + 
        geom_pointrange() + 
        scale_x_continuous(trans = "log10", name = "Sequence Depth") +
        ylab("Richness") +
        facet_wrap(~Subplot_Descriptions, scales = 'free') +
        theme(legend.text = element_text(hjust = 0)) +
        labs(title = "16S Rarefaction Curves") +
        theme_bw()

curve_16S_facet

curve_16S <- ggplot(rarefaction_curve_data_summary_verbose %>% filter(Measure == "Observed"), aes(x = Depth, y = Alpha_diversity_mean,  ymin = Alpha_diversity_mean - Alpha_diversity_sd, ymax = Alpha_diversity_mean + Alpha_diversity_sd, colour = Subplot_Descriptions, group = Sample)) + 
        geom_line(alpha=1) + 
        geom_pointrange() + 
        scale_x_continuous(trans = "log10", name = "Sequence Depth") +
        ylab("Richness") +
        #facet_wrap(~Subplot_Descriptions, scales = 'free') +
        theme(legend.text = element_text(hjust = 0)) +
        labs(title = "16S Rarefaction Curves") +
        theme_bw()

curve_16S
```

#Taxonomy - Stacked Bar Plots

Rarefy data and setting up to make stacked bar plots
```{r}
# Rarefy data to 1000 reads 
set.seed(01221990)
rare_16S <- rarefy_even_depth(data_16S_counts, sample.size = 1000)
rare_16S_1 <- rarefy_even_depth(data_16S_counts, sample.size = min(sample_sums(data_16S_counts)))

data_16S_rel <- transform_sample_counts(rare_16S, function(x) x/sum(x))

# subset to non insecticide plots
data_16S_rel_noinsect <- subset_samples(data_16S_rel, Subplot_Descriptions%in%c("ambient", "irrigated_control", "warmed_drought", "drought", "warmed"))
summarize_phyloseq(data_16S_rel_noinsect)

treatment_names <- c("ambient" = "Ambient",
                     "drought" = "Drought",
                     "irrigated_control" = "Irrigated \n Control",
                     "warmed" = "Warmed",
                     "warmed_drought" = "Warmed \n Drought", 
                     "drought_insecticide" = "Drought \n Insecticide", 
                     "insecticide" = "Insecticide", 
                     'warmed_drought_insecticide' = "Warmed \n Drought \n Insecticide", 
                     'warmed_insecticide' = "Warmed \n Insecticide")

treatment_names_noinsect <- c("ambient" = "Ambient",
                     "drought" = "Drought",
                     "irrigated_control" = "Irrigated \n Control",
                     "warmed" = "Warmed",
                     "warmed_drought" = "Warmed \n Drought")
```

#Phylum Level
```{r}
phylum_sum_16S <- tapply(taxa_sums(data_16S_rel), 
                         tax_table(data_16S_rel)[, "Phylum"], sum, na.rm=TRUE)
top10Phylum_16S <- names(sort(phylum_sum_16S, TRUE))[1:10]

abundant_16S_phylum <- subset_taxa(data_16S_rel, Phylum %in% top10Phylum_16S)
abundant_16S_phylum_glom <- tax_glom(abundant_16S_phylum, taxrank = "Phylum")
abundant_16S_phylum_melt <- psmelt(abundant_16S_phylum_glom)

sum2 <- abundant_16S_phylum_melt %>% 
  group_by(Subplot_Descriptions, Drought, Phylum) %>%
  summarise(means = mean(Abundance))

#Everything that's left is considered rare  
rare <- sum2 %>%
  group_by(Subplot_Descriptions, Drought) %>%
  summarise(means = 1- sum(means)) %>%
  mutate(Phylum = "Rare Taxa")

#concatenate the datasets
sum2 = rbind(sum2, rare)

#order groups
sum2$Phylum <- forcats::fct_relevel(sum2$Phylum, "Rare Taxa", after = Inf)

# Stacked bar plot for top phylas
# facet by timing of soil collection (aka "Drought")
#png("T7_warmx_16S_phylum_stacked_bar_plot_by_drought.png", units="in", width=10, height=6, res=300)
ggplot(sum2, aes(x = Subplot_Descriptions, y = means, fill = Phylum)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Drought, scale="free") +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
                                     "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought",
                                  "drought_insecticide" = "Drought Insecticide", 
                                  "insecticide" = "Insecticide", 
                                  'warmed_drought_insecticide' = "Warmed Drought Insecticide", 
                                  'warmed_insecticide' = "Warmed Insecticide")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
#dev.off()

# facet by treatment
#png("T7_warmx_16S_phylum_stacked_bar_plot_by_treatment.png", units="in", width=10, height=6, res=300)
ggplot(sum2, aes(x = Drought, y = means, fill = Phylum)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
                                     "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
         scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought",
                                  "drought_insecticide" = "Drought Insecticide", 
                                  "insecticide" = "Insecticide", 
                                  'warmed_drought_insecticide' = "Warmed Drought Insecticide", 
                                  'warmed_insecticide' = "Warmed Insecticide")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 10, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
#dev.off()

```

#Phylum Level - non insecticide plots
```{r}
phylum_sum_16S_noinsect <- tapply(taxa_sums(data_16S_rel_noinsect), 
                         tax_table(data_16S_rel_noinsect)[, "Phylum"], sum, na.rm=TRUE)
top10Phylum_16S_noinsect <- names(sort(phylum_sum_16S_noinsect, TRUE))[1:10]

abundant_16S_phylum_noinsect <- subset_taxa(data_16S_rel_noinsect, Phylum %in% top10Phylum_16S_noinsect)
abundant_16S_phylum_glom_noinsect <- tax_glom(abundant_16S_phylum_noinsect, taxrank = "Phylum")
abundant_16S_phylum_melt_noinsect <- psmelt(abundant_16S_phylum_glom_noinsect)

sum2_noinsect <- abundant_16S_phylum_melt_noinsect %>% 
  group_by(Subplot_Descriptions, Drought, Phylum) %>%
  summarise(means = mean(Abundance))

#Everything that's left is considered rare  
rare <- sum2_noinsect %>%
  group_by(Subplot_Descriptions, Drought) %>%
  summarise(means = 1- sum(means)) %>%
  mutate(Phylum = "Rare Taxa")

#concatenate the datasets
sum2_noinsect = rbind(sum2_noinsect, rare)

#order groups
sum2_noinsect$Phylum <- forcats::fct_relevel(sum2_noinsect$Phylum, "Rare Taxa", after = Inf)

# Stacked bar plot for top phylas
# facet by timing of soil collection (aka "Drought")
#png("T7_warmx_16S_phylum_stacked_bar_plot_by_drought.png", units="in", width=10, height=6, res=300)
ggplot(sum2_noinsect, aes(x = Subplot_Descriptions, y = means, fill = Phylum)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Drought, scale="free") +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
                                     "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
#dev.off()

# facet by treatment
png("T7_warmx_16S_phylum_stacked_bar_plot_by_treatment_noinsect.png", units="in", width=10, height=6, res=300)
ggplot(sum2_noinsect, aes(x = Drought, y = means, fill = Phylum)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names_noinsect)) +
        scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1", "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
         scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 12),
              strip.text = element_text(size = 12, face = "italic"), 
              axis.title = element_blank(),
              title = element_text(size = 12),
              legend.position = "right", 
              legend.title = element_text(size = 12),
              legend.text = element_text(size = 12),
              axis.line = element_blank(),
              axis.text.y=element_text(size=12))
dev.off()

```

#Class Level - non insecticide
```{r}
# 16S Habitat Overview Plot
# Rarefy data to 1000 reads 
set.seed(01221990)

class_sum_16S_noinsect <- tapply(taxa_sums(data_16S_rel_noinsect), 
                         tax_table(data_16S_rel_noinsect)[, "Class"], sum, na.rm=TRUE)
top15Class_16S_noinsect <- names(sort(class_sum_16S_noinsect, TRUE))[1:15]

abundant_16S_class_noinsect <- subset_taxa(data_16S_rel_noinsect, Class %in% top15Class_16S_noinsect)
abundant_16S_class_glom_noinsect <- tax_glom(abundant_16S_class_noinsect, taxrank = "Class")
abundant_16S_class_melt_noinsect <- psmelt(abundant_16S_class_glom_noinsect)

sum5 <- abundant_16S_class_melt_noinsect %>% 
  group_by(Subplot_Descriptions, Drought, Class) %>%
  summarise(means = mean(Abundance))

#Everything that's left is considered rare  
rare <- sum5 %>%
  group_by(Subplot_Descriptions, Drought) %>%
  summarise(means = 1- sum(means)) %>%
  mutate(Class = "Rare Taxa")

#concatenate the datasets
sum5 = rbind(sum5, rare)

#order groups
sum5$Class <- forcats::fct_relevel(sum5$Class, "Rare Taxa", after = Inf)

# Stacked bar plot for top genus'
# facet_grid by timing of soil sampling "Drought"
#png("T7_warmx_16S_class_stacked_bar_plot.png", units="in", width=10, height=6, res=300)
ggplot(sum5, aes(x = Subplot_Descriptions, y = means, fill = Class)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Drought, scale="free") +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1", "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4", "cyan2", "deeppink", "darkslategrey", "gold", "lightslateblue")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
#dev.off()

# facet_grid by treatment aka "Subplot_Descriptions"
png("T7_warmx_16S_class_stacked_bar_plot_by_treatment_noinsect.png", units="in", width=10, height=6, res=300)
ggplot(sum5, aes(x = Drought, y = means, fill = Class)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Subplot_Descriptions, scale="free", labeller = 
                           as_labeller(treatment_names_noinsect)) +
        scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1", "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4", "cyan2", "deeppink", "darkslategrey", "gold", "lightslateblue")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1, size = 12),
              strip.text = element_text(size = 12, face = "italic"), 
              axis.title = element_blank(),
              title = element_text(size = 12),
              legend.position = "right", 
              legend.title = element_text(size = 12),
              legend.text = element_text(size = 12),
              axis.line = element_blank(),
              axis.text.y=element_text(size=12))
dev.off()
```

#Family Level
```{r}
# 16S Habitat Overview Plot
# Rarefy data to 1000 reads 
set.seed(01221990)

family_sum_16S <- tapply(taxa_sums(data_16S_rel), 
                         tax_table(data_16S_rel)[, "Family"], sum, na.rm=TRUE)
top10Family_16S <- names(sort(family_sum_16S, TRUE))[1:10]

abundant_16S_family <- subset_taxa(data_16S_rel, Family %in% top10Family_16S)
abundant_16S_family_glom <- tax_glom(abundant_16S_family, taxrank = "Family")
abundant_16S_family_melt <- psmelt(abundant_16S_family_glom)

sum3 <- abundant_16S_family_melt %>% 
  group_by(Subplot_Descriptions, Drought, Family) %>%
  summarise(means = mean(Abundance))

#Everything that's left is considered rare  
rare <- sum3 %>%
  group_by(Subplot_Descriptions, Drought) %>%
  summarise(means = 1- sum(means)) %>%
  mutate(Family = "Rare Taxa")

#concatenate the datasets
sum3 = rbind(sum3, rare)

#order groups
sum3$Family <- forcats::fct_relevel(sum3$Family, "Rare Taxa", after = Inf)

# Stacked bar plot for top phylas
#png("T7_warmx_16S_family_stacked_bar_plot.png", units="in", width=10, height=6, res=300)
ggplot(sum3, aes(x = Subplot_Descriptions, y = means, fill = Family)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Drought, scale="free") +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        #scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
        #                             "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought",
                                  "drought_insecticide" = "Drought Insecticide", 
                                  "insecticide" = "Insecticide", 
                                  'warmed_drought_insecticide' = "Warmed Drought Insecticide", 
                                  'warmed_insecticide' = "Warmed Insecticide")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
#dev.off()

# facet_grid by treatment aka "Subplot_Descriptions"
#png("T7_warmx_16S_family_stacked_bar_plot_by_treatment.png", units="in", width=10, height=6, res=300)
ggplot(sum3, aes(x = Drought, y = means, fill = Family)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        #scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
        #                             "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought",
                                  "drought_insecticide" = "Drought Insecticide", 
                                  "insecticide" = "Insecticide", 
                                  'warmed_drought_insecticide' = "Warmed Drought Insecticide", 
                                  'warmed_insecticide' = "Warmed Insecticide")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
#dev.off()
```

#Family Level - non insecticide
```{r}
# 16S Habitat Overview Plot
# Rarefy data to 1000 reads 
set.seed(01221990)

family_sum_16S_noinsect <- tapply(taxa_sums(data_16S_rel_noinsect), 
                         tax_table(data_16S_rel_noinsect)[, "Family"], sum, na.rm=TRUE)
top20Family_16S_noinsect <- names(sort(family_sum_16S_noinsect, TRUE))[1:20]

abundant_16S_family_noinsect <- subset_taxa(data_16S_rel_noinsect, Family %in% top20Family_16S_noinsect)
abundant_16S_family_glom_noinsect <- tax_glom(abundant_16S_family_noinsect, taxrank = "Family")
abundant_16S_family_melt_noinsect <- psmelt(abundant_16S_family_glom_noinsect)

sum3_noinsect <- abundant_16S_family_melt_noinsect %>% 
  group_by(Subplot_Descriptions, Drought, Family) %>%
  summarise(means = mean(Abundance))

#Everything that's left is considered rare  
rare <- sum3_noinsect %>%
  group_by(Subplot_Descriptions, Drought) %>%
  summarise(means = 1- sum(means)) %>%
  mutate(Family = "Rare Taxa")

#concatenate the datasets
sum3_noinsect = rbind(sum3_noinsect, rare)

#order groups
sum3$Family <- forcats::fct_relevel(sum3$Family, "Rare Taxa", after = Inf)

# stacked bar plot for top Families
# facet_grid by treatment aka "Subplot_Descriptions"
#png("T7_warmx_16S_family_stacked_bar_plot_by_treatment.png", units="in", width=10, height=6, res=300)
ggplot(sum3_noinsect, aes(x = Drought, y = means, fill = Family)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        #scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
        #                             "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought",
                                  "drought_insecticide" = "Drought Insecticide", 
                                  "insecticide" = "Insecticide", 
                                  'warmed_drought_insecticide' = "Warmed Drought Insecticide", 
                                  'warmed_insecticide' = "Warmed Insecticide")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
#dev.off()
```

#Genus Level
```{r}
# 16S Habitat Overview Plot
# Rarefy data to 1000 reads 
set.seed(01221990)

genus_sum_16S <- tapply(taxa_sums(data_16S_rel), 
                         tax_table(data_16S_rel)[, "Genus"], sum, na.rm=TRUE)
top10Genus_16S <- names(sort(genus_sum_16S, TRUE))[1:10]

abundant_16S_genus <- subset_taxa(data_16S_rel, Genus %in% top10Genus_16S)
abundant_16S_genus_glom <- tax_glom(abundant_16S_genus, taxrank = "Genus")
abundant_16S_genus_melt <- psmelt(abundant_16S_genus_glom)

sum4 <- abundant_16S_genus_melt %>% 
  group_by(Subplot_Descriptions, Drought, Genus) %>%
  summarise(means = mean(Abundance))

#Everything that's left is considered rare  
rare <- sum4 %>%
  group_by(Subplot_Descriptions, Drought) %>%
  summarise(means = 1- sum(means)) %>%
  mutate(Genus = "Rare Taxa")

#concatenate the datasets
sum4 = rbind(sum4, rare)

#order groups
sum4$Genus <- forcats::fct_relevel(sum4$Genus, "Rare Taxa", after = Inf)

# Stacked bar plot for top genus'
# facet_grid by timing of soil sampling "Drought"
#png("T7_warmx_16S_genus_stacked_bar_plot.png", units="in", width=10, height=6, res=300)
ggplot(sum4, aes(x = Subplot_Descriptions, y = means, fill = Genus)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Drought, scale="free") +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        #scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
        #                             "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought",
                                  "drought_insecticide" = "Drought Insecticide", 
                                  "insecticide" = "Insecticide", 
                                  'warmed_drought_insecticide' = "Warmed Drought Insecticide", 
                                  'warmed_insecticide' = "Warmed Insecticide")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
#dev.off()

# facet_grid by treatment aka "Subplot_Descriptions"
#png("T7_warmx_16S_genus_stacked_bar_plot_by_treatment.png", units="in", width=11, height=6, res=300)
ggplot(sum4, aes(x = Drought, y = means, fill = Genus)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        #scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
        #                             "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought",
                                  "drought_insecticide" = "Drought Insecticide", 
                                  "insecticide" = "Insecticide", 
                                  'warmed_drought_insecticide' = "Warmed Drought Insecticide", 
                                  'warmed_insecticide' = "Warmed Insecticide")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
#dev.off()
```

#Genus Level - non insecticide
```{r}
# 16S Habitat Overview Plot
# Rarefy data to 1000 reads 
set.seed(01221990)

genus_sum_16S_noinsect <- tapply(taxa_sums(data_16S_rel_noinsect), 
                         tax_table(data_16S_rel_noinsect)[, "Genus"], sum, na.rm=TRUE)
top20Genus_16S_noinsect <- names(sort(genus_sum_16S_noinsect, TRUE))[1:20]

abundant_16S_genus_noinsect <- subset_taxa(data_16S_rel_noinsect, Genus %in% top20Genus_16S_noinsect)
abundant_16S_genus_glom_noinsect <- tax_glom(abundant_16S_genus_noinsect, taxrank = "Genus")
abundant_16S_genus_melt_noinsect <- psmelt(abundant_16S_genus_glom_noinsect)

sum4_noinsect <- abundant_16S_genus_melt_noinsect %>% 
  group_by(Subplot_Descriptions, Drought, Genus) %>%
  summarise(means = mean(Abundance))

#Everything that's left is considered rare  
rare <- sum4_noinsect %>%
  group_by(Subplot_Descriptions, Drought) %>%
  summarise(means = 1- sum(means)) %>%
  mutate(Genus = "Rare Taxa")

#concatenate the datasets
sum4_noinsect = rbind(sum4_noinsect, rare)

#order groups
sum4_noinsect$Genus <- forcats::fct_relevel(sum4_noinsect$Genus, "Rare Taxa", after = Inf)

# Stacked bar plot for top genus'
# facet_grid by treatment aka "Subplot_Descriptions"
#png("T7_warmx_16S_genus_stacked_bar_plot_by_treatment.png", units="in", width=11, height=6, res=300)
ggplot(sum4_noinsect, aes(x = Drought, y = means, fill = Genus)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        #scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
        #                             "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought",
                                  "drought_insecticide" = "Drought Insecticide", 
                                  "insecticide" = "Insecticide", 
                                  'warmed_drought_insecticide' = "Warmed Drought Insecticide", 
                                  'warmed_insecticide' = "Warmed Insecticide")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
#dev.off()
```

#Create phyloseq object that doesn't include insecticide treatments with filtered data
```{r}
data_16S_filtered_noinsect <- subset_samples(data_16S_filtered, Subplot_Descriptions%in%c("ambient", "irrigated_control", "warmed_drought", "drought", "warmed"))
summarize_phyloseq(data_16S_filtered_noinsect)

# just peak drought
data_16S_filtered_nona <- subset_samples(data_16S_filtered_noinsect, !is.na(Drought))
data_16S_filtered_noinsect_peakdrought <- subset_samples(data_16S_filtered_nona, Drought == "Peak-Drought")
summarize_phyloseq(data_16S_filtered_noinsect_peakdrought)

# removing just pre drought
data_16S_filtered_nopredrought <- subset_samples(data_16S_filtered_noinsect, Drought%in%c("Peak-Drought", "Post-Drought", "Recovery"))
summarize_phyloseq(data_16S_filtered_nopredrought)

# removing irrigated control
data_16S_filtered_noIRC <- subset_samples(data_16S_filtered, Subplot_Descriptions%in%c("ambient", "warmed_drought", "drought", "warmed"))
summarize_phyloseq(data_16S_filtered_noIRC)

# peak drought and no irrigated control
data_16S_filtered_peak_noIRC <- subset_samples(data_16S_filtered_noIRC, Drought == "Peak-Drought")
```

#Alpha Diversity
```{r}
# Rarefy the samples without replacement. Rarefaction is used to simulate even number of reads per sample. The rarefaction depth chosen is the 90% of the minimum sample depth in the dataset
ps.rarefied = rarefy_even_depth(data_16S_filtered_noinsect, rngseed=1, sample.size = 0.9*min(sample_sums(data_16S_filtered_noinsect)), replace=F)

ps.rarefied_1 = rarefy_even_depth(data_16S_filtered_noinsect, rngseed=1, sample.size = min(sample_sums(data_16S_filtered_noinsect)), replace=F)

# Plotting Alpha Diversity
treatment_names <- c("ambient" = "Ambient",
                     "drought" = "Drought",
                     "irrigated_control" = "Irrigated Control",
                     "warmed" = "Warmed",
                     "warmed_drought" = "Warmed + Drought")

plot_richness(ps.rarefied, x="Drought", color="Subplot_Descriptions", measures=c("Observed"))
plot_richness(ps.rarefied, x="Drought", measures=c("Observed", "Shannon")) + geom_boxplot()
plot_richness(ps.rarefied, x="Subplot_Descriptions", measures=c("Observed")) + geom_boxplot() + facet_grid(.~Drought)
plot_richness(ps.rarefied, x="Subplot_Descriptions", measures=c("Shannon")) + geom_boxplot() + facet_grid(.~Drought)

png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(ps.rarefied, x="Drought", measures=c("Observed")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(y = "Observed OTUs", x = "Time of Sampling") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
dev.off()

#png("T7_warmx_16S_shannon_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(ps.rarefied, x="Drought", measures=c("Shannon")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(x = "Time of Sampling", y = "Shannon Index") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()

richness <- estimate_richness(ps.rarefied)
head(richness)
kruskal.test(richness$Shannon ~ sample_data(ps.rarefied)$Subplot_Descriptions)
pairwise.wilcox.test(richness$Shannon, sample_data(ps.rarefied)$Subplot_Descriptions, p.adj = "bonf")

kruskal.test(richness$Shannon ~ sample_data(ps.rarefied)$Drought)
pairwise.wilcox.test(richness$Shannon, sample_data(ps.rarefied)$Drought, p.adj = "bonf")
```

#Beta Diversity

Working outside of a phyloseq object
```{r}
otu.test <- data.frame(ps.rarefied_1@otu_table@.Data)
taxa.test <- data.frame(ps.rarefied_1@tax_table@.Data)


```

Exploring with just peak drought & no irrigated control (removing irrigated control gets you a balanced design)
```{r}
# rarefy samples
ps.rarefied_peak_noIRC = rarefy_even_depth(data_16S_filtered_peak_noIRC, rngseed=1, sample.size = min(sample_sums(data_16S_filtered_peak_noIRC)), replace=F)

# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(ps.rarefied_peak_noIRC, method="bray")
ordination_pcoa = ordinate(ps.rarefied_peak_noIRC, method="PCoA", distance=bray_dist)

# https://stackoverflow.com/questions/76465183/plot-ordination-function-of-phyloseq-plots-shapes-double-within-itself
# Save the PCoA vectors data from your ordinated object
vectors <- as.data.frame(ordination_pcoa$vectors)
values <- as.data.frame(ordination_pcoa$values)

library(tibble)
vectors <- rownames_to_column(vectors, "SampleID")

metadata <- rownames_to_column(data.frame(ps.rarefied_peak_noIRC@sam_data), "SampleID")
vectors_metadata <- dplyr::inner_join(vectors, metadata, by = "SampleID")

ggplot(vectors_metadata, aes(x = Axis.1, y = Axis.2, color = Subplot_Descriptions)) + geom_point()

#png("T7_warmx_16S_pcoa_no_insects.png", units="in", width=10, height=6, res=300)
plot_ordination(ps.rarefied_peak_noIRC, ordination_pcoa, color="Subplot_Descriptions") + 
        geom_point(aes(color = Subplot_Descriptions), size=4, alpha=0.75) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        scale_colour_manual(name = "Subplot Descriptions", 
                          breaks = c("ambient", "warmed", "drought", "warmed_drought"),
                          labels = c("Ambient", "Warmed", "Drought", "Warmed + Drought"),
                          values = c("ambient" = "darkblue", "drought" = "gray58", "warmed" = "red2", "warmed_drought" = "darkred")) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
#dev.off()

# Test whether the treatments ("Subplot_Descriptions") differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(ps.rarefied_peak_noIRC)$Subplot_Descriptions)

# trying Jennifer's code below:
#bray.dist <- phyloseq::distance(ps.rarefied, method = "bray”)

#set.seed(53)

# setting up the permutation object
#perm1 <- with(sample_data(ps.rarefied), how(blocks = Replicate, plots = Plots(strata = Footprint, #type="free”), nperm = 9999))

#adonis2(wunifrac_dist ~ sample_data(ps.rarefied)$Footprint*sample_data(ps.rarefied)$Subplot, by = #"margin”, permutations = #perm1) 
#If the interaction is not significant, you should rerun the model without the interaction. For this code, if the interaction is not significant, it won’t show you if the main effects are significant. 

perm1 <- with(sample_data(ps.rarefied_peak_noIRC), how(blocks = Replicate, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))

adonis2(bray_dist ~ sample_data(ps.rarefied_peak_noIRC)$Footprint * sample_data(ps.rarefied_peak_noIRC)$Subplot, by = "margin", permutations = perm1)

adonis2(bray_dist ~ sample_data(ps.rarefied_peak_noIRC)$Footprint + sample_data(ps.rarefied_peak_noIRC)$Subplot, by = "margin", permutations = perm1)

adonis2(bray_dist ~ sample_data(ps.rarefied_peak_noIRC)$Subplot_Descriptions + sample_data(ps.rarefied_peak_noIRC)$Footprint + sample_data(ps.rarefied_peak_noIRC)$Subplot, by = "margin", permutations = perm1)

###############################################################################
ordination_rda = ordinate(ps.rarefied_peak_noIRC, method="RDA", distance=bray_dist)

rda_scores <- vegan::scores(ordination_rda, display = "sites")
# Add labels, make a data.frame
rda_scores <- data.frame(labels = rownames(arrowmat), arrowmat)
# Define the arrow aesthetic mapping
arrow_map = aes(xend = CA1, yend = CA2, x = 0, y = 0, shape = NULL, color = NULL, 
    label = labels)
label_map = aes(x = 1.2 * CA1, y = 1.2 * CA2, shape = NULL, color = NULL, 
    label = labels)

arrowhead = arrow(length = unit(0.05, "npc"))

plot_ordination(ps.rarefied_peak_noIRC, ordination_rda, color="Subplot_Descriptions") + 
        geom_point(aes(color = Subplot_Descriptions), size=4, alpha=0.75) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        scale_colour_manual(name = "Subplot Descriptions", 
                          breaks = c("ambient", "warmed", "drought", "warmed_drought"),
                          labels = c("Ambient", "Warmed", "Drought", "Warmed + Drought"),
                          values = c("ambient" = "darkblue", "drought" = "gray58", "warmed" = "red2", "warmed_drought" = "darkred")) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1) +
        geom_segment(arrow_map, size = 0.5, data = rda_scores, color = "gray", arrow = arrowhead) + 
        geom_text(label_map, size = 2, data = rda_scores)

###############################################################################
# https://github.com/joey711/phyloseq/issues/274

ordination_cca = ordinate(ps.rarefied_peak_noIRC, method="CCA")

# Mimic biplot from example
p0 = plot_ordination(ps.rarefied_peak_noIRC, ordination_cca, type = "biplot", color = "Subplot_Descriptions")
p0

# Now add the environmental variables as arrows
arrowmat = vegan::scores(ordination_cca, display = "sites")
# Add labels, make a data.frame
arrowdf <- data.frame(labels = rownames(arrowmat), arrowmat)
# Define the arrow aesthetic mapping
arrow_map = aes(xend = CA1, yend = CA2, x = 0, y = 0, shape = NULL, color = NULL, 
    label = labels)
label_map = aes(x = 1.2 * CA1, y = 1.2 * CA2, shape = NULL, color = NULL, 
    label = labels)
# Make a new graphic
arrowhead = arrow(length = unit(0.05, "npc"))
p1 = p0 + 
        geom_segment(arrow_map, size = 0.5, data = arrowdf, color = "gray", arrow = arrowhead) + 
        geom_text(label_map, size = 2, data = arrowdf)
p1
```


PCoA
```{r}
# Weighted-Unifrac takes into account the relative abundance of species/taxa shared between samples, whereas 
# unweighted-Unifrac only considers presence/absence

# PCoA plot using UNWEIGHTED UniFrac as distance
wunifrac_dist = phyloseq::distance(ps.rarefied_1, method="unifrac", weighted=F)
ordination = ordinate(ps.rarefied_1, method="PCoA", distance=wunifrac_dist)

#png("T7_warmx_16S_pcoa_no_insects.png", units="in", width=10, height=6, res=300)
plot_ordination(ps.rarefied_1, ordination, color="Subplot_Descriptions", shape = "Drought") + 
        geom_point(aes(color = Subplot_Descriptions, shape = Drought), size=4, alpha=0.75) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        scale_colour_manual(name = "Subplot Descriptions", 
                          breaks = c("irrigated_control", "ambient", "warmed", "drought", "warmed_drought"),
                          labels = c("Irrigated Control", "Ambient", "Warmed", "Drought", "Warmed + Drought"),
                          values = c("ambient" = "darkblue", "drought" = "gray58", "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        labs(x="Axis 1",
             y="Axis 2") +
        theme(aspect.ratio=1)
#dev.off()

# PCoA plot using WEIGHTED UniFrac as distance
weighted_unifrac_dist <- phyloseq::distance(ps.rarefied_1, method="unifrac", weighted=T)
ordination <- ordinate(ps.rarefied_1, method="PCoA", distance=weighted_unifrac_dist)

#png("T7_warmx_16S_pcoa_weighted_unifrac_no_insects.png", units="in", width=10, height=6, res=300)
plot_ordination(ps.rarefied_1, ordination, color="Subplot_Descriptions", shape = "Drought") + 
        geom_point(aes(color = Subplot_Descriptions, shape = Drought), size=4, alpha=0.75) +
        #stat_ellipse() +
        scale_colour_manual(name = "Subplot Descriptions", 
                          breaks = c("irrigated_control", "ambient", "warmed", "drought", "warmed_drought"),
                          labels = c("Irrigated Control", "Ambient", "Warmed", "Drought", "Warmed + Drought"),
                          values = c("ambient" = "darkblue", "drought" = "gray58", "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
#dev.off()

# Test whether the treatments ("Subplot_Descriptions") differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(wunifrac_dist ~ sample_data(ps.rarefied_1)$Subplot_Descriptions + sample_data(ps.rarefied_1)$Drought)

# Test whether the treatments ("Subplot_Descriptions") differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(weighted_unifrac_dist ~ sample_data(ps.rarefied)$Subplot_Descriptions + sample_data(ps.rarefied)$Drought)

# trying Jennifer's code below:
#bray.dist <- phyloseq::distance(ps.rarefied, method = "bray”)

#set.seed(53)

# setting up the permutation object
#perm1 <- with(sample_data(ps.rarefied), how(blocks = Replicate, plots = Plots(strata = Footprint, #type="free”), nperm = 9999))

#adonis2(wunifrac_dist ~ sample_data(ps.rarefied)$Footprint*sample_data(ps.rarefied)$Subplot, by = #"margin”, permutations = #perm1) 
#If the interaction is not significant, you should rerun the model without the interaction. For this code, if the interaction is not significant, it won’t show you if the main effects are significant. 

perm1 <- with(sample_data(ps.rarefied_1), how(blocks = Replicate, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))

adonis2(wunifrac_dist ~ sample_data(ps.rarefied_1)$Footprint * sample_data(ps.rarefied_1)$Subplot, by = "margin", permutations = perm1)

```


RDA
```{r}
# RDA plot using WEIGHTED UniFrac as distance
RDA_ps.rarefied = ordinate(ps.rarefied, method="RDA")

plot_ordination(ps.rarefied, RDA_ps.rarefied, color="Subplot_Descriptions", shape = "Drought") + 
        geom_point(aes(color = Subplot_Descriptions, shape = Drought), size=4, alpha=0.75) +
        #stat_ellipse() +
        scale_colour_manual(name = "Subplot Descriptions", 
                          breaks = c("irrigated_control", "ambient", "warmed", "drought", "warmed_drought"),
                          labels = c("Irrigated Control", "Ambient", "Warmed", "Drought", "Warmed + Drought"),
                          values = c("ambient" = "darkblue", "drought" = "gray58", "irrigated_control" = "deepskyblue",
                                     "warmed" = "red2", "warmed_drought" = "darkred")) +
        
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)


```

PCoA excluding irrigated control
```{r}
ps.rarefied_noIRC = rarefy_even_depth(data_16S_filtered_noIRC, rngseed=1, sample.size = 0.9*min(sample_sums(data_16S_filtered_nopredrought)), replace=F)

# PCoA plot using WEIGHTED UniFrac as distance
weighted_unifrac_dist = phyloseq::distance(ps.rarefied_noIRC, method="unifrac", weighted=T)
ordination = ordinate(ps.rarefied_nopredrought, method="PCoA", distance=weighted_unifrac_dist)

#png("T7_warmx_16S_pcoa_weighted_unifrac_no_insects.png", units="in", width=10, height=6, res=300)
plot_ordination(ps.rarefied_noIRC, ordination, color="Subplot_Descriptions", shape = "Drought") + 
        geom_point(aes(color = Subplot_Descriptions, shape = Drought), size=4, alpha=0.75) +
        #stat_ellipse() +
        scale_colour_manual(name = "Subplot Descriptions", 
                          breaks = c("irrigated_control", "ambient", "warmed", "drought", "warmed_drought"),
                          labels = c("Irrigated Control", "Ambient", "Warmed", "Drought", "Warmed + Drought"),
                          values = c("ambient" = "darkblue", "drought" = "gray58", "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
#dev.off()

# Test whether the treatments ("Subplot_Descriptions") differ significantly from each other using PERMANOVA:
set.seed(22)
adonis2(weighted_unifrac_dist ~ sample_data(ps.rarefied_noIRC)$Subplot_Descriptions + sample_data(ps.rarefied_noIRC)$Drought)

perm1 <- with(sample_data(ps.rarefied_noIRC), how(blocks = Replicate, plots = Plots(strata=Footprint, type="free"), nperm = 9999))

adonis2(wunifrac_dist ~ sample_data(ps.rarefied_noIRC)$Footprint * sample_data(ps.rarefied_noIRC)$Subplot, by = "margin", permutations = perm1)
```


PCoA excluding irrigated control
```{r}
ps.rarefied_noIRC = rarefy_even_depth(data_16S_filtered_noIRC, rngseed=1, sample.size = 0.9*min(sample_sums(data_16S_filtered_nopredrought)), replace=F)

# PCoA plot using WEIGHTED UniFrac as distance
weighted_unifrac_dist = phyloseq::distance(ps.rarefied_noIRC, method="unifrac", weighted=T)
ordination = ordinate(ps.rarefied_nopredrought, method="PCoA", distance=weighted_unifrac_dist)

#png("T7_warmx_16S_pcoa_weighted_unifrac_no_insects.png", units="in", width=10, height=6, res=300)
plot_ordination(ps.rarefied_noIRC, ordination, color="Subplot_Descriptions", shape = "Drought") + 
        geom_point(aes(color = Subplot_Descriptions, shape = Drought), size=4, alpha=0.75) +
        #stat_ellipse() +
        scale_colour_manual(name = "Subplot Descriptions", 
                          breaks = c("irrigated_control", "ambient", "warmed", "drought", "warmed_drought"),
                          labels = c("Irrigated Control", "Ambient", "Warmed", "Drought", "Warmed + Drought"),
                          values = c("ambient" = "darkblue", "drought" = "gray58", "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
#dev.off()

# Test whether the treatments ("Subplot_Descriptions") differ significantly from each other using PERMANOVA:
set.seed(22)
adonis2(weighted_unifrac_dist ~ sample_data(ps.rarefied_noIRC)$Subplot_Descriptions + sample_data(ps.rarefied_noIRC)$Drought)

perm1 <- with(sample_data(ps.rarefied_noIRC), how(blocks = Replicate, plots = Plots(strata=Footprint, type="free"), nperm = 9999))

adonis2(wunifrac_dist ~ sample_data(ps.rarefied_noIRC)$Footprint * sample_data(ps.rarefied_noIRC)$Subplot, by = "margin", permutations = perm1)
```

# NMDS - Ordination using Phyloseq package - excluding pre drought
```{r}
ord_16S <- prune_taxa(names(sort(taxa_sums(ps.rarefied_nopredrought), TRUE)[1:50]), ps.rarefied_nopredrought)

# bray curtis ordination
bray_pcoa_16S <- ordinate(
        physeq = ord_16S, 
        method = "NMDS", 
        distance = "bray"
        )

# plot ordination
plot_ordination(ord_16S, bray_pcoa_16S, "samples", color = "Subplot_Descriptions")
# Run 20 stress 0.137421
#png("T7_warmx_16S_braycurtis_NMDS_ordination.png", units="in", width=10, height=6, res=300)
plot_ordination(
  physeq = ord_16S,                                                         #phyloseq object
  ordination = bray_pcoa_16S) +                                                #ordination
  geom_point(aes(fill = Subplot_Descriptions, shape = Drought), size = 3) + #sets fill color to sampletype
        stat_ellipse(aes(fill = factor(Subplot_Descriptions)), geom = "polygon", alpha = .5) +
        scale_fill_manual(name = "Subplot Descriptions", 
                          breaks = c("irrigated_control", "ambient", "warmed", "drought", "warmed_drought"),
                          labels = c("Irrigated Control", "Ambient", "Warmed", "Drought", "Warmed + Drought"),
                          values = c("ambient" = "darkblue", "drought" = "gray58", "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.title = element_blank())                                      #removes legend title
#dev.off()

# NMDS and ANOSIM test
dist = phyloseq::distance(ps.rarefied_nopredrought, method="bray", binary = TRUE)
ordination = ordinate(ps.rarefied_nopredrought, method="NMDS", distance=dist)
plot_ordination(ps.rarefied_nopredrought, ordination, color="Subplot_Descriptions") +
        theme_classic() +
        theme(strip.background = element_blank())

metadata <- data.frame(sample_data(ord_16S))
anosim(dist, metadata$Subplot_Descriptions)
anosim(dist, metadata$Drought)
```

#Differential abundance testing

# DESeq2 - warmed vs drought
```{r}
# Convert phyloseq data to DESeq2 dds (dds = DESeq2 Data Set) object
# Note - cannot use phyloseq object that has sample data ordered or the line of code will not work
# Re line 102
# x <- factor( x , ordered = FALSE )
sample_data(data_16S_filtered_nopredrought) <- factor(sample_data(data_16S_filtered_nopredrought), ordered = FALSE)
dds <- phyloseq_to_deseq2(data_16S_filtered_nopredrought, ~ Subplot_Descriptions + Drought)
dds <- DESeq(dds, 
             fitType="parametric")
alpha = 0.01
res = results(dds)
summary(res)
# whatever you list first for the contrasts, is going to be on the top of your ggplot that you make below
res = results(dds, contrast = c("Subplot_Descriptions", "drought", "warmed"), alpha=alpha)
res = res[order(res$padj, na.last=NA), ]
res_sig = res[(res$padj < alpha), ]
res_sig = cbind(as(res_sig, "data.frame"), as(tax_table(data_16S_filtered_noinsect)[rownames(res_sig), ], "matrix"))
head(res_sig)
dim(res_sig) # [1] 196  15

# Creates a data frame from results
df <- as.data.frame(res)
# Adds taxon column that includes names of taxa
df$taxon <- rownames(df)
# Orders the rows of data frame in increasing order firstly based on column "log2FoldChange" and secondly based on "padj" column
# log2FoldChange = The effect size estimate. This value indicates how much the gene or transcript's expression seems to have changed between the comparison and control groups. This value is reported on a logarithmic scale to base 2
df <- df %>% arrange(log2FoldChange, padj)

knitr::kable(head(df)) %>% 
  kableExtra::kable_styling("striped") %>% 
  kableExtra::scroll_box(width = "100%")

# summary plot
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)
}
# Genus order
x = tapply(res_sig$log2FoldChange, res_sig$Genus, function(x) max(x))
x = sort(x, TRUE)
res_sig$Genus = factor(as.character(res_sig$Genus), levels=names(x))

png("diff_abun_warmed_drought.png", units="in", width=14, height=7, res=300)
ggplot(res_sig, aes(x=Genus, y=log2FoldChange, color=Phylum)) + 
        geom_point(size=3) + 
        geom_hline(yintercept=0, linewidth=1) +
        annotate("text", x = 80, y=15, label = "Drought", size=4) +
        annotate("text", x = 80, y=-15, label = "Warmed", size=4) +
        scale_colour_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
                                     "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4", "aquamarine", "darkmagenta", "gold", "mediumblue", "salmon", "deepskyblue")) +
        theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
dev.off()
```

Code from Brandon (similar to above)
# DIFFERENTIAL ABUNDANCE: drought vs warmed
```{r}
## DIFFERENTIAL ABUNDANCE: drought vs warmed

# just subset data at peak drought?
# bact_peakdrought <- subset_samples(data_16S_filtered_noinsect, Drought == "Peak_Drought")

diagdds <- phyloseq_to_deseq2(data_16S_filtered_nopredrought, ~ Subplot_Descriptions + Drought) 
# calculate geometric means prior to estimate size factors
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(diagdds), 1, gm_mean)
diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)
diagdds = DESeq(diagdds, fitType="parametric")
################################################################################
results <- results(diagdds, alpha = 0.01, contrast = c("Subplot_Descriptions", "drought", "warmed"))
sigtab_results <- results[which(results$padj < 0.01), ]
#sigtab_results_drought_vs_warmed <- results[which(results$padj < 0.01), ]
sigtab_results_TAX <- cbind(as(sigtab_results, "data.frame"), as(tax_table(data_16S_filtered_noinsect)[row.names(sigtab_results), ], "matrix"))
sigtab_results_TAX_genra <- subset(sigtab_results_TAX, !is.na(Genus))

# Order Genus 
x <- tapply(sigtab_results_TAX_genra$log2FoldChange, sigtab_results_TAX_genra$Class, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra$Class <- factor(as.character(sigtab_results_TAX_genra$Class), levels=names(x))
# Genus order
x <- tapply(sigtab_results_TAX_genra$log2FoldChange, sigtab_results_TAX_genra$Genus, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra$Genus <- factor(as.character(sigtab_results_TAX_genra$Genus), levels=names(x))
# Family
x <- tapply(sigtab_results_TAX_genra$log2FoldChange, sigtab_results_TAX_genra$Family, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra$Family <- factor(as.character(sigtab_results_TAX_genra$Family), levels=names(x))

# Creates a data frame from results
df1 <- as.data.frame(sigtab_results_TAX_genra)
# Adds taxon column that includes names of taxa
df1$taxon <- rownames(sigtab_results_TAX_genra)
# Orders the rows of data frame in increasing order firstly based on column "log2FoldChange" and secondly based on "padj" column
# log2FoldChange = The effect size estimate. This value indicates how much the gene or transcript's expression seems to have changed between the comparison and control groups. This value is reported on a logarithmic scale to base 2
df1 <- df1 %>% arrange(log2FoldChange, padj)

knitr::kable(head(df1)) %>% 
  kableExtra::kable_styling("striped") %>% 
  kableExtra::scroll_box(width = "100%")

png("diff_abun_warmed_drought_1.png", units="in", width=14, height=7, res=300)
ggplot(sigtab_results_TAX_genra, aes(x=Class, y=log2FoldChange, color=Phylum)) + 
        geom_point(size=3) + 
        geom_hline(yintercept=0, linewidth=1) +
        annotate("text", x = 40, y=15, label = "Drought", size=4) +
        annotate("text", x = 40, y=-5, label = "Warmed", size=4) +
        scale_colour_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
                                     "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4", "aquamarine", "darkmagenta", "gold", "mediumblue", "salmon", "deepskyblue", "darkseagreen", "gray0", "darkslategrey", "firebrick2")) +
        theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
dev.off()
```

# DIFFERENTIAL ABUNDANCE - ambient vs drought
```{r}
results1 <- results(diagdds, alpha = 0.01, contrast = c("Subplot_Descriptions", "ambient", "drought"))
sigtab_results1 <- results1[which(results1$padj < 0.01), ]
#sigtab_results_ambient_vs_drought <- results[which(results$padj < 0.01), ]
sigtab_results_TAX1 <- cbind(as(sigtab_results1, "data.frame"), as(tax_table(data_16S_filtered_noinsect)[row.names(sigtab_results1), ], "matrix"))
sigtab_results_TAX_genra1 <- subset(sigtab_results_TAX1, !is.na(Genus))

# Order Genus 
x <- tapply(sigtab_results_TAX_genra1$log2FoldChange, sigtab_results_TAX_genra1$Class, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra1$Class <- factor(as.character(sigtab_results_TAX_genra1$Class), levels=names(x))
# Genus order
x <- tapply(sigtab_results_TAX_genra1$log2FoldChange, sigtab_results_TAX_genra1$Genus, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra1$Genus <- factor(as.character(sigtab_results_TAX_genra1$Genus), levels=names(x))
# Family
x <- tapply(sigtab_results_TAX_genra1$log2FoldChange, sigtab_results_TAX_genra1$Family, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra1$Family <- factor(as.character(sigtab_results_TAX_genra1$Family), levels=names(x))

# Creates a data frame from results
df2 <- as.data.frame(sigtab_results_TAX_genra1)
# Adds taxon column that includes names of taxa
df2$taxon <- rownames(sigtab_results_TAX_genra1)
# Orders the rows of data frame in increasing order firstly based on column "log2FoldChange" and secondly based on "padj" column
# log2FoldChange = The effect size estimate. This value indicates how much the gene or transcript's expression seems to have changed between the comparison and control groups. This value is reported on a logarithmic scale to base 2
df2 <- df2 %>% arrange(log2FoldChange, padj)

knitr::kable(head(df2)) %>% 
  kableExtra::kable_styling("striped") %>% 
  kableExtra::scroll_box(width = "100%")

png("diff_abun_ambient_drought.png", units="in", width=14, height=7, res=300)
ggplot(sigtab_results_TAX_genra1, aes(x=Class, y=log2FoldChange, color=Phylum)) + 
        geom_point(size=3) + 
        geom_hline(yintercept=0, linewidth=1) +
        annotate("text", x = 10, y=15, label = "Ambient", size=4) +
        annotate("text", x = 10, y=-5, label = "Drought", size=4) +
        scale_colour_manual(values = c("firebrick4","tomato", "olivedrab1", "orange1" ,"deepskyblue", "khaki1",
                                     "forestgreen", "darkorange3", "midnightblue", "powderblue", "mediumorchid4", "aquamarine", "darkmagenta", "gold", "mediumblue", "salmon", "goldenrod4", "darkseagreen", "gray0", "darkslategrey", "firebrick2")) +
        theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
dev.off()
```

# DIFFERENTIAL ABUNDANCE - ambient vs warmed
```{r}
results2 <- results(diagdds, alpha = 0.01, contrast = c("Subplot_Descriptions", "ambient", "warmed"))
sigtab_results2 <- results2[which(results2$padj < 0.01), ]
#sigtab_results_ambient_vs_warmed <- results[which(results$padj < 0.01), ]
sigtab_results_TAX2 <- cbind(as(sigtab_results2, "data.frame"), as(tax_table(data_16S_filtered_noinsect)[row.names(sigtab_results2), ], "matrix"))
sigtab_results_TAX_genra2 <- subset(sigtab_results_TAX2, !is.na(Genus))

# Order Genus 
x <- tapply(sigtab_results_TAX_genra2$log2FoldChange, sigtab_results_TAX_genra2$Class, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra2$Class <- factor(as.character(sigtab_results_TAX_genra2$Class), levels=names(x))
# Genus order
x <- tapply(sigtab_results_TAX_genra2$log2FoldChange, sigtab_results_TAX_genra2$Genus, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra2$Genus <- factor(as.character(sigtab_results_TAX_genra2$Genus), levels=names(x))
# Family
x <- tapply(sigtab_results_TAX_genra2$log2FoldChange, sigtab_results_TAX_genra2$Family, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra2$Family <- factor(as.character(sigtab_results_TAX_genra2$Family), levels=names(x))

# Creates a data frame from results
df3 <- as.data.frame(sigtab_results_TAX_genra2)
# Adds taxon column that includes names of taxa
df3$taxon <- rownames(sigtab_results_TAX_genra2)
# Orders the rows of data frame in increasing order firstly based on column "log2FoldChange" and secondly based on "padj" column
# log2FoldChange = The effect size estimate. This value indicates how much the gene or transcript's expression seems to have changed between the comparison and control groups. This value is reported on a logarithmic scale to base 2
df3 <- df3 %>% arrange(log2FoldChange, padj)

knitr::kable(head(df3)) %>% 
  kableExtra::kable_styling("striped") %>% 
  kableExtra::scroll_box(width = "100%")

png("diff_abun_ambient_warmed.png", units="in", width=14, height=7, res=300)
ggplot(sigtab_results_TAX_genra2, aes(x=Class, y=log2FoldChange, color=Phylum)) + 
        geom_point(size=3) + 
        geom_hline(yintercept=0, linewidth=1) +
        annotate("text", x = 20, y=10, label = "Ambient", size=4) +
        annotate("text", x = 20, y=-5, label = "Warmed", size=4) +
        scale_colour_manual(values = c("firebrick4","tomato", "olivedrab1", "orange1" ,"deepskyblue", "khaki1",
                                     "forestgreen", "darkorange3", "midnightblue", "powderblue", "mediumorchid4", "aquamarine", "darkmagenta", "gold", "mediumblue", "salmon", "goldenrod4", "darkseagreen", "gray0", "darkslategrey", "firebrick2", "lightslateblue")) +
        theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
dev.off()
```

# DIFFERENTIAL ABUNDANCE - ambient vs warmed + drought
```{r}
results3 <- results(diagdds, alpha = 0.01, contrast = c("Subplot_Descriptions", "ambient", "warmed_drought"))
sigtab_results3 <- results3[which(results3$padj < 0.01), ]
#sigtab_results_ambient_vs_warmed <- results[which(results$padj < 0.01), ]
sigtab_results_TAX3 <- cbind(as(sigtab_results3, "data.frame"), as(tax_table(data_16S_filtered_noinsect)[row.names(sigtab_results3), ], "matrix"))
sigtab_results_TAX_genra3 <- subset(sigtab_results_TAX3, !is.na(Genus))

# Order Genus 
x <- tapply(sigtab_results_TAX_genra3$log2FoldChange, sigtab_results_TAX_genra3$Class, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra3$Class <- factor(as.character(sigtab_results_TAX_genra3$Class), levels=names(x))
# Genus order
x <- tapply(sigtab_results_TAX_genra3$log2FoldChange, sigtab_results_TAX_genra3$Genus, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra3$Genus <- factor(as.character(sigtab_results_TAX_genra3$Genus), levels=names(x))
# Family
x <- tapply(sigtab_results_TAX_genra3$log2FoldChange, sigtab_results_TAX_genra3$Family, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra3$Family <- factor(as.character(sigtab_results_TAX_genra3$Family), levels=names(x))

# Creates a data frame from results
df4 <- as.data.frame(sigtab_results_TAX_genra3)
# Adds taxon column that includes names of taxa
df4$taxon <- rownames(sigtab_results_TAX_genra3)
# Orders the rows of data frame in increasing order firstly based on column "log2FoldChange" and secondly based on "padj" column
# log2FoldChange = The effect size estimate. This value indicates how much the gene or transcript's expression seems to have changed between the comparison and control groups. This value is reported on a logarithmic scale to base 2
df4 <- df4 %>% arrange(log2FoldChange, padj)

knitr::kable(head(df4)) %>% 
  kableExtra::kable_styling("striped") %>% 
  kableExtra::scroll_box(width = "100%")

png("diff_abun_ambient_warmed_drought.png", units="in", width=14, height=7, res=300)
ggplot(sigtab_results_TAX_genra3, aes(x=Class, y=log2FoldChange, color=Phylum)) + 
        geom_point(size=3) + 
        geom_hline(yintercept=0, linewidth=1) +
        annotate("text", x = 20, y=15, label = "Ambient", size=4) +
        annotate("text", x = 20, y=-5, label = "Warmed + Drought", size=4) +
        scale_colour_manual(values = c("firebrick4","tomato", "olivedrab1", "orange1" ,"deepskyblue", "khaki1",
                                     "forestgreen", "darkorange3", "midnightblue", "powderblue", "mediumorchid4", "aquamarine", "darkmagenta", "gold", "mediumblue", "salmon", "goldenrod4", "darkseagreen", "gray0", "darkslategrey", "firebrick2", "springgreen", "thistle1", "turquoise4")) +
        theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
dev.off()
```

# DIFFERENTIAL ABUNDANCE: ambient vs drought
# PEAK DROUGHT ONLY
```{r}
## DIFFERENTIAL ABUNDANCE: drought vs warmed

diagdds1 <- phyloseq_to_deseq2(data_16S_filtered_noinsect_peakdrought, ~ Subplot_Descriptions) 
# calculate geometric means prior to estimate size factors
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(diagdds1), 1, gm_mean)
diagdds1 = estimateSizeFactors(diagdds1, geoMeans = geoMeans)
diagdds1 = DESeq(diagdds1, fitType="parametric")
results4 <- results(diagdds1, alpha = 0.01, contrast = c("Subplot_Descriptions", "ambient", "drought"))
sigtab_results4 <- results4[which(results4$padj < 0.01), ]
#sigtab_results_drought_vs_warmed <- results[which(results$padj < 0.01), ]
sigtab_results_TAX4 <- cbind(as(sigtab_results4, "data.frame"), as(tax_table(data_16S_filtered_noinsect_peakdrought)[row.names(sigtab_results4), ], "matrix"))
sigtab_results_TAX_genra4 <- subset(sigtab_results_TAX4, !is.na(Genus))

# Order Genus 
x <- tapply(sigtab_results_TAX_genra4$log2FoldChange, sigtab_results_TAX_genra4$Class, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra4$Class <- factor(as.character(sigtab_results_TAX_genra4$Class), levels=names(x))
# Genus order
x <- tapply(sigtab_results_TAX_genra4$log2FoldChange, sigtab_results_TAX_genra4$Genus, function(x) max(x))
x <- sort(x, TRUE)
sigtab_results_TAX_genra4$Genus <- factor(as.character(sigtab_results_TAX_genra4$Genus), levels=names(x))

# Creates a data frame from results
df5 <- as.data.frame(sigtab_results_TAX_genra4)
# Adds taxon column that includes names of taxa
df5$taxon <- rownames(sigtab_results_TAX_genra4)
# Orders the rows of data frame in increasing order firstly based on column "log2FoldChange" and secondly based on "padj" column
# log2FoldChange = The effect size estimate. This value indicates how much the gene or transcript's expression seems to have changed between the comparison and control groups. This value is reported on a logarithmic scale to base 2
df5 <- df5 %>% arrange(log2FoldChange, padj)

knitr::kable(head(df5)) %>% 
  kableExtra::kable_styling("striped") %>% 
  kableExtra::scroll_box(width = "100%")

#png("diff_abun_warmed_drought_1.png", units="in", width=14, height=7, res=300)
ggplot(sigtab_results_TAX_genra4, aes(x=Class, y=log2FoldChange, color=Phylum)) + 
        geom_point(size=3) + 
        geom_hline(yintercept=0, linewidth=1) +
        #annotate("text", x = 40, y=15, label = "Drought", size=4) +
        #annotate("text", x = 40, y=-5, label = "Warmed", size=4) +
        scale_colour_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
                                     "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4", "aquamarine", "darkmagenta", "gold", "mediumblue", "salmon", "deepskyblue", "darkseagreen", "gray0", "darkslategrey", "firebrick2")) +
        theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))
#dev.off()
```


```{r}
# trying something different from above
# https://rstudio-pubs-static.s3.amazonaws.com/187480_55b8817c975341c0bbc81934d773e768.html
res1 <- results(dds, tidy=TRUE, contrast=c("Subplot_Descriptions", "warmed", "drought")) %>%
  arrange(padj, pvalue) %>%
  tbl_df()
# contrast argument from above is comparing "warmed" and "drought" from "subplot_descriptions"
res1

knitr::kable(head(res1)) %>% 
  kableExtra::kable_styling("striped") %>% 
  kableExtra::scroll_box(width = "100%")

taxa_of_interest <- res1$row[1:12]
stopifnot(all(taxa_of_interest %in% names(dds)))
taxa_of_interest

tcounts <- t(log2((counts(dds[taxa_of_interest, ], normalized=TRUE, replaced=FALSE)+.5))) %>%
  merge(colData(dds), ., by="row.names") %>%
  gather(taxa, relative_abun, (ncol(.)-length(taxa_of_interest)+1):ncol(.))

tcounts %>% 
  select(Row.names, Subplot_Descriptions, Drought, taxa, relative_abun) %>% 
  head %>% 
  knitr::kable()

colnames(taxonomy)[colnames(taxonomy) == "X"] ="taxa"
tcounts1 <- merge(tcounts, taxonomy)

ggplot(tcounts1, aes(Subplot_Descriptions, relative_abun, fill=Drought)) + 
  geom_boxplot() + 
  facet_wrap(~Phylum, scales="free_y") + 
  labs(x="Treatment", 
       y="Relative Abundance (log normalized counts)", 
       fill="Drought")
```

Differential abundance using ANCOM - this doesn't work below, I think bc the package has been updated
```{r}
# Differentially Abundant ASVs due to "Drought" with "Subplot_Descriptions" covariate

meta_data = data_16S_filtered@sam_data
meta_data <- as.data.frame(meta_data)
meta_data$names <- rownames(meta_data)

feature_table = data_16S_filtered@otu_table
feature_table <- as.data.frame(feature_table)

sample_var = "names" ; group_var = NULL
out_cut = 0.05; zero_cut = 0.90; lib_cut = 1000; neg_lb = FALSE
prepro = feature_table_pre_process(feature_table, meta_data, sample_var, group_var, 
                                   out_cut, zero_cut, lib_cut, neg_lb)
feature_table = prepro$feature_table # Preprocessed feature table
meta_data = prepro$meta_data # Preprocessed metadata
struc_zero = prepro$structure_zeros # Structural zero info

# Step 2: ANCOM

main_var = "Drought"; p_adj_method = "BH"; alpha = 0.05
adj_formula = "Subplot_Descriptions"; rand_formula = NULL
t_start = Sys.time()
res_drought = ANCOM(feature_table, meta_data, struc_zero, main_var, p_adj_method, 
                          alpha, adj_formula, rand_formula)
t_end = Sys.time()
t_run = t_end - t_start

###################################################################################3
pseq_genus <- phyloseq::tax_glom(data_16S_filtered, taxrank = "Genus")
# The taxonomy table
tax_mat = as(tax_table(pseq_genus), "matrix")

pseq = makePhyloseqFromTreeSummarizedExperiment(pseq_genus)
out = ancom(data = NULL, assay_name = NULL,
            tax_level = "Genus", phyloseq = pseq,
            p_adj_method = "holm", prv_cut = 0.10, lib_cut = 1000,
            main_var = "Drought", adj_formula = "Subplot_Descriptions",
            rand_formula = NULL, lme_control = NULL, struc_zero = TRUE,
            neg_lb = TRUE, alpha = 0.05, n_cl = 2)
res <- out$res

knitr::kable(head(res$diff_abn)) %>% kableExtra::kable_styling("striped") %>% 
  kableExtra::scroll_box(width = "100%")
```

# Differential abundance using lefse
```{r}
library(microbiomeMarker)
lefse <- run_lefse(data_16S_filtered_noinsect,
                  wilcoxon_cutoff = 0.01,
                  norm = "CPM",
                  group = "Drought",
                  kw_cutoff = 0.01,
                  multigrp_strat = TRUE,
                  lda_cutoff = 4
)


head(marker_table(lefse))

# bar plot
plot_ef_bar(lefse)
# dot plot
plot_ef_dot(lefse)
```

#Heat Map
```{r}
set.seed(01221990)

#top100ITSsklearn <- names(sort(taxa_sums(ITSsklearn), decreasing=TRUE))[1:20]
#ITSsklearn.top20 <- transform_sample_counts(ITSsklearn, function(OTU) OTU / sum(OTU))
#ITSsklearn.top20 <- prune_taxa(top20ITSsklearn, ITSsklearn.top20)

set.seed(01221990)
rare_16S <- rarefy_even_depth(data_16S_counts, sample.size = 1000)
data_16S_rel <- transform_sample_counts(rare_16S, function(x) x/sum(x))
top30Genus_16S <- names(sort(genus_sum_16S, TRUE))[1:30]
abundant_16S_genus_top30 <- subset_taxa(data_16S_rel, Genus %in% top30Genus_16S)
abundant_16S_genus_top30_glom <- tax_glom(abundant_16S_genus_top30, taxrank = "Genus")

abundant_16S_genus_top30_melt <- psmelt(abundant_16S_genus_top30_glom)
abundant_16S_genus_sum <- abundant_16S_genus_top30_melt %>% 
        group_by(Subplot_Descriptions, Drought, Genus) %>% 
        summarize(avg_abundance = mean(Abundance, na.rm = TRUE),
                  se = std.error(Abundance, na.rm = TRUE))
sum5 <- abundant_16S_genus_top30_melt %>% 
  group_by(Subplot_Descriptions, Drought, Genus) %>%
  summarise(means = mean(Abundance))

# code below doesn't work
heat.map <- plot_taxa_heatmap(abundant_16S_genus,
  subset.top = 20,
  VariableA = "Subplot_Descriptions",
  taxonomic.level = "Genus",
  #heatcolors = colorRampPalette(rev(brewer.pal(n = 7, name = "RdYlBu")))(100),
  transformation = "log10"
)

# code below works but isn't what I want
abundance_heatmap(abundant_16S_genus_top30_glom, treatment = c("Drought"), subset = NULL,
classification = 'Genus', transformation = 'log', colors = 'default',
treatment_labels = NULL, sample_labels = NULL, classification_labels= NULL)

plot_heatmap(abundant_16S_genus_top30_glom, "unifrac", taxa.label = "Genus", sample.label = "Drought", weighted=TRUE)

heatmap(sum4)
```

Peak-Drought Heat Map
```{r}
data_16S_filtered_noinsect_peakdrought
set.seed(01221990)
rare_16S <- rarefy_even_depth(data_16S_filtered_noinsect_peakdrought, sample.size = 1000)
data_16S_rel <- transform_sample_counts(rare_16S, function(x) x/sum(x))
top30Genus_16S <- names(sort(genus_sum_16S, TRUE))[1:30]
abundant_16S_genus_top30 <- subset_taxa(data_16S_rel, Genus %in% top30Genus_16S)
abundant_16S_genus_top30_glom <- tax_glom(abundant_16S_genus_top30, taxrank = "Genus")

# this one looks the best 
abundance_heatmap(abundant_16S_genus_top30_glom, treatment = c("Drought"), subset = NULL,
classification = 'Genus', transformation = 'log', colors = 'default',
treatment_labels = NULL, sample_labels = NULL, classification_labels= NULL)

plot_heatmap(abundant_16S_genus_top30_glom, taxa.label = "Genus", sample.label = NULL, weighted=TRUE)
```


```{r}
mine.heatmap <- ggplot(data = abundant_16S_genus_sum, mapping = aes(x = Subplot_Descriptions,
                                                       y = Genus,
                                                       fill = avg_abundance)) +
  geom_tile() +
  xlab(label = "Treatment")

mine.heatmap
```

