---
title: "T7 warmx 16S Analysis 2021 & 2023"
author: "Moriah Young"
date: "2024-07-02"
output: pdf_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, echo=FALSE, message=FALSE}
rm(list = ls()) # clear working environment

# load packages
library(compositions)
library(cowplot)
library(cluster)
library(data.table)
library(devtools)
library(plyr)
library(dplyr)
library(doParallel)
library(factoextra)
library(ggplot2)
library(ggsignif)
library(gridExtra)
library(hillR)
library(microbiome)
library(multcomp)
library(nlme)
library(phyloseq)
library(qiime2R)
library(tidyverse)
library(vegan)
library(zCompositions)
library(permute)
library(pheatmap)
library(plotrix)
library(microViz)
#library(phylosmith)
#library(DESeq2)
#library(ANCOMBC)

# Set working directory from .Renviron
dir <- Sys.getenv("DATA_DIR")
list.files(dir)
```


```{r}
# 2021 data
# Create phyloseq objects of filtered data (16S)
phyloseq_2021 <- read_csv2phyloseq(otu.file = "/Users/moriahyoung/Desktop/phyloseq_object/16S-2021-merged-table-filtered.csv",
                                       taxonomy.file = "/Users/moriahyoung/Desktop/phyloseq_object/16S-2021-taxonomy-filtered.csv",
                                       metadata.file = "/Users/moriahyoung/Desktop/phyloseq_object/bacterial_2021_metadata.csv")
summarize_phyloseq(phyloseq_2021)

# 2023 data
# Create phyloseq objects of filtered data (16S)
phyloseq_2023 <- read_csv2phyloseq(otu.file = "/Users/moriahyoung/Desktop/phyloseq_object/16S-2023-table-filtered-noncontam.csv",
                                       taxonomy.file = "/Users/moriahyoung/Desktop/phyloseq_object/16S-2023-taxonomy-filtered-noncontam.csv",
                                       metadata.file = "/Users/moriahyoung/Desktop/phyloseq_object/bacterial_2023_metadata.csv")
summarize_phyloseq(phyloseq_2023)

# merge 2021 and 2023 phyloseq objects together
phyloseq_all <- merge_phyloseq(phyloseq_2021, phyloseq_2023)

samdat_tbl(phyloseq_all)
summarize_phyloseq(phyloseq_all)
```

#Filter data and rarefy data
```{r}
# filter data
# remove samples that have less than 1000 reads
data_16S_counts <- prune_samples(sample_sums(phyloseq_all)>=1000, phyloseq_all)
summarize_phyloseq(data_16S_counts)
# remove singletons
data_16S_counts1 <- filter_taxa(data_16S_counts, function (x) {sum(x > 0) > 1}, prune=TRUE)
summarize_phyloseq(data_16S_counts1)

# Rarefy data to min sample size reads 
set.seed(1022)
rare_16S <- rarefy_even_depth(data_16S_counts1, sample.size = min(sample_sums(data_16S_counts1)))
summarize_phyloseq(rare_16S)

# Need to change "Year" column to a factor
# Access the sample data
rare_16S_sample_data <- sample_data(rare_16S)

# Convert a specific column to a factor (e.g., "Year")
rare_16S_sample_data$Year <- as.factor(rare_16S_sample_data$Year)

# Update the phyloseq object with the modified sample data
sample_data(rare_16S) <- rare_16S_sample_data

# Check the structure to confirm
str(sample_data(rare_16S))

###########################################
# did not run code below
# Remove taxa not seen more than 3 times in at least 20% of the samples. This protects against an OTU with small mean & trivially large C.V.
phyloseq_all_filtered <- filter_taxa(phyloseq_all, function(x) sum(x > 3) > (0.2*length(x)), TRUE)
summarize_phyloseq(phyloseq_all_filtered)

# Rarefy data to min sample size reads 
set.seed(1022)
#rare_16S <- rarefy_even_depth(data_16S_counts, sample.size = 1000)
rare_16S <- rarefy_even_depth(phyloseq_all, sample.size = min(sample_sums(phyloseq_all)))
summarize_phyloseq(rare_16S)
sample_data(rare_16S)
```

# PEAK DROUGHT NO IRRIGATED CONTROL
#Subsetting phyloseq object
```{r}
# no insecticide treatment and no irrigated control
data_rare_noinsect <- subset_samples(rare_16S, Subplot_Descriptions%in%c("ambient", "warmed_drought", "drought", "warmed"))

# peak drought only
data_rare_noinsect_nona <- subset_samples(data_rare_noinsect, !is.na(sampling_period))
data_rare_noinsect_peak <- subset_samples(data_rare_noinsect_nona, sampling_period == "Peak-Drought")
summarize_phyloseq(data_rare_noinsect_peak)
```

#Alpha Diversity
```{r}
# order sample data - "sampling_period" and "Subplot_Descriptions"
sample_data(data_rare_noinsect_peak)$sampling_period <- ordered(sample_data(data_rare_noinsect_peak)$sampling_period, c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"))
sample_data(data_rare_noinsect_peak)$Subplot_Descriptions <- ordered(sample_data(data_rare_noinsect_peak)$Subplot_Descriptions, c( "ambient", "warmed","drought", "warmed_drought"))

# Plotting Alpha Diversity
treatment_names <- c("ambient" = "Ambient",
                     "drought" = "Drought",
                     "warmed" = "Warmed",
                     "warmed_drought" = "Warmed + Drought")

plot_richness(data_rare_noinsect_peak, x="Year", color="Subplot_Descriptions", measures=c("Observed"))
plot_richness(data_rare_noinsect_peak, x="Year", measures=c("Observed", "Shannon")) + geom_boxplot()
plot_richness(data_rare_noinsect_peak, x="Subplot_Descriptions", measures=c("Observed")) + geom_boxplot() + facet_grid(.~Year)
plot_richness(data_rare_noinsect_peak, x="Subplot_Descriptions", measures=c("Shannon")) + geom_boxplot() + facet_grid(.~Year)


png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_peak, x="Year", measures=c("Observed")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(y = "Observed OTUs", x = "Time of Sampling") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
dev.off()

#png("T7_warmx_16S_shannon_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_peak, x="Year", measures=c("Shannon")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(x = "Time of Sampling", y = "Shannon Index") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()

# analysis of alpha diversity
richness <- estimate_richness(data_rare_noinsect_peak)
head(richness)
kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_peak)$Subplot_Descriptions)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_peak)$Subplot_Descriptions, p.adj = "bonf")

kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_peak)$Year)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_peak)$Year, p.adj = "bonf")
```

Beta Diversity
```{r}
# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(data_rare_noinsect_peak, method="bray")
ordination_pcoa = ordinate(data_rare_noinsect_peak, method="PCoA", distance=bray_dist)

# https://stackoverflow.com/questions/76465183/plot-ordination-function-of-phyloseq-plots-shapes-double-within-itself
# Save the PCoA vectors data from your ordinated object
vectors_peak <- as.data.frame(ordination_pcoa$vectors)
values_peak <- as.data.frame(ordination_pcoa$values)

library(tibble)
vectors_peak <- rownames_to_column(vectors_peak, "SampleID")

metadata_peak <- rownames_to_column(data.frame(data_rare_noinsect_peak@sam_data), "SampleID")
vectors_metadata_peak <- dplyr::inner_join(vectors_peak, metadata_peak, by = "SampleID")

ggplot(vectors_metadata_peak, aes(x = Axis.1, y = Axis.2, color = Year)) + geom_point()

p1 <- plot_ordination(data_rare_noinsect_peak, ordination_pcoa, color="Year") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p1

p2 <- plot_ordination(data_rare_noinsect_peak, ordination_pcoa, axes = c(1, 3), color="Year") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p2

# Extract sample data
sample_data_df <- as.data.frame(sample_data(data_rare_noinsect_peak))

# Extract site scores from the ordination object
site_scores <- ordination_pcoa$vectors

# Combine scores with sample data
ordination_df <- data.frame(site_scores, sample_data_df)

# Plot the ordination using ggplot2
p <- ggplot(ordination_df, aes_string(x = "Axis.1", y = "Axis.2", color = "Year")) +
  geom_point(size = 4) +
  stat_ellipse(aes(group = Year), type = "norm") + # Add ellipses
  theme_minimal()
p

# Test whether the sampling periods ("Dates") differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_peak)$Year)

# trying Jennifer's code below:
#bray.dist <- phyloseq::distance(ps.rarefied, method = "bray”)

#set.seed(53)

#adonis2(wunifrac_dist ~ sample_data(ps.rarefied)$Footprint*sample_data(ps.rarefied)$Subplot, by = #"margin”, permutations = #perm1) 
#If the interaction is not significant, you should rerun the model without the interaction. For this code, if the interaction is not significant, it won’t show you if the main effects are significant. 

# setting up the permutation object
perm1 <- with(sample_data(data_rare_noinsect_peak), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))

perm2 <- with(sample_data(data_rare_noinsect_peak), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_noinsect_peak)$Year, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_peak)$Year, by = "margin", permutations = perm2)
```


# POST DROUGHT NO IRRIGATED CONTROL
#Subsetting phyloseq object
```{r}
# post drought only
data_rare_noinsect_post <- subset_samples(data_rare_noinsect_nona, sampling_period == "Post-Drought")
summarize_phyloseq(data_rare_noinsect_post)
```

#Alpha Diversity
```{r}
# order sample data - "Subplot_Descriptions"
sample_data(data_rare_noinsect_post)$Subplot_Descriptions <- ordered(sample_data(data_rare_noinsect_post)$Subplot_Descriptions, c( "ambient", "warmed","drought", "warmed_drought"))

plot_richness(data_rare_noinsect_post, x="Year", color="Subplot_Descriptions", measures=c("Observed"))
plot_richness(data_rare_noinsect_post, x="Year", measures=c("Observed", "Shannon")) + geom_boxplot()
plot_richness(data_rare_noinsect_post, x="Subplot_Descriptions", measures=c("Observed")) + geom_boxplot() + facet_grid(.~Year)
plot_richness(data_rare_noinsect_peak, x="Subplot_Descriptions", measures=c("Shannon")) + geom_boxplot() + facet_grid(.~Year)


#png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_post, x="Year", measures=c("Observed")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(y = "Observed OTUs", x = "Year") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
#dev.off()

#png("T7_warmx_16S_shannon_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_post, x="Year", measures=c("Shannon")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(x = "Year", y = "Shannon Index") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()

# analysis of alpha diversity
richness <- estimate_richness(data_rare_noinsect_post)
head(richness)
kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_post)$Subplot_Descriptions)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_post)$Subplot_Descriptions, p.adj = "bonf")

kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_post)$Year)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_post)$Year, p.adj = "bonf")
```

Beta Diversity
```{r}
# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(data_rare_noinsect_post, method="bray")
ordination_pcoa = ordinate(data_rare_noinsect_post, method="PCoA", distance=bray_dist)

# https://stackoverflow.com/questions/76465183/plot-ordination-function-of-phyloseq-plots-shapes-double-within-itself
# Save the PCoA vectors data from your ordinated object
vectors_post <- as.data.frame(ordination_pcoa$vectors)
values_post <- as.data.frame(ordination_pcoa$values)

library(tibble)
vectors_post <- rownames_to_column(vectors_post, "SampleID")

metadata_post <- rownames_to_column(data.frame(data_rare_noinsect_post@sam_data), "SampleID")
vectors_metadata_post <- dplyr::inner_join(vectors_post, metadata_post, by = "SampleID")

ggplot(vectors_metadata_post, aes(x = Axis.1, y = Axis.2, color = Year)) + geom_point()

p1 <- plot_ordination(data_rare_noinsect_post, ordination_pcoa, color="Year") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p1

p2 <- plot_ordination(data_rare_noinsect_post, ordination_pcoa, axes = c(1, 3), color="Year") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p2

# Extract sample data
sample_data_df <- as.data.frame(sample_data(data_rare_noinsect_post))

# Extract site scores from the ordination object
site_scores <- ordination_pcoa$vectors

# Combine scores with sample data
ordination_df <- data.frame(site_scores, sample_data_df)

# Plot the ordination using ggplot2
p <- ggplot(ordination_df, aes_string(x = "Axis.1", y = "Axis.2", color = "Year")) +
  geom_point(size = 4) +
  stat_ellipse(aes(group = Year), type = "norm") + # Add ellipses
  theme_minimal()
p

# Test whether the sampling periods ("Dates") differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_post)$Year)

# trying Jennifer's code below:
#bray.dist <- phyloseq::distance(ps.rarefied, method = "bray”)

#set.seed(53)

#adonis2(wunifrac_dist ~ sample_data(ps.rarefied)$Footprint*sample_data(ps.rarefied)$Subplot, by = #"margin”, permutations = #perm1) 
#If the interaction is not significant, you should rerun the model without the interaction. For this code, if the interaction is not significant, it won’t show you if the main effects are significant. 

# setting up the permutation object
perm1 <- with(sample_data(data_rare_noinsect_post), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))
# dropping footprint in model
perm2 <- with(sample_data(data_rare_noinsect_post), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_noinsect_post)$Year, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_post)$Year, by = "margin", permutations = perm2)
```

# 2023
#Subsetting phyloseq object
```{r}
# post drought only
data_rare_noinsect_2023 <- subset_samples(data_rare_noinsect_nona, Year == "2023")
summarize_phyloseq(data_rare_noinsect_2023)
```

#Alpha Diversity
```{r}
# order sample data - "Subplot_Descriptions"
sample_data(data_rare_noinsect_2023)$Subplot_Descriptions <- ordered(sample_data(data_rare_noinsect_2023)$Subplot_Descriptions, c( "ambient", "warmed","drought", "warmed_drought"))
# "sampling_period"
sample_data(data_rare_noinsect_2023)$sampling_period <- ordered(sample_data(data_rare_noinsect_2023)$sampling_period, c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"))

plot_richness(data_rare_noinsect_2023, x="sampling_period", color="Subplot_Descriptions", measures=c("Observed"))
plot_richness(data_rare_noinsect_2023, x="sampling_period", measures=c("Observed", "Shannon")) + geom_boxplot()
plot_richness(data_rare_noinsect_2023, x="Subplot_Descriptions", measures=c("Observed")) + geom_boxplot() + facet_grid(.~sampling_period)
plot_richness(data_rare_noinsect_2023, x="Subplot_Descriptions", measures=c("Shannon")) + geom_boxplot() + facet_grid(.~sampling_period)


#png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_2023, x="sampling_period", measures=c("Observed")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(y = "Observed OTUs", x = "Sampling Period") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
#dev.off()

#png("T7_warmx_16S_shannon_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_2023, x="sampling_period", measures=c("Shannon")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(x = "Year", y = "Shannon Index") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()

# analysis of alpha diversity
richness <- estimate_richness(data_rare_noinsect_2023)
head(richness)
kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_2023)$Subplot_Descriptions)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_post)$Subplot_Descriptions, p.adj = "bonf")

kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_2023)$sampling_period)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_post)$sampling_period, p.adj = "bonf")
```

Beta Diversity
```{r}
# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(data_rare_noinsect_2023, method="bray")
ordination_pcoa = ordinate(data_rare_noinsect_2023, method="PCoA", distance=bray_dist)

# https://stackoverflow.com/questions/76465183/plot-ordination-function-of-phyloseq-plots-shapes-double-within-itself
# Save the PCoA vectors data from your ordinated object
vectors_2023 <- as.data.frame(ordination_pcoa$vectors)
values_2023 <- as.data.frame(ordination_pcoa$values)

library(tibble)
vectors_2023 <- rownames_to_column(vectors_2023, "SampleID")

metadata_2023 <- rownames_to_column(data.frame(data_rare_noinsect_2023@sam_data), "SampleID")
vectors_metadata_2023 <- dplyr::inner_join(vectors_2023, metadata_2023, by = "SampleID")

ggplot(vectors_metadata_2023, aes(x = Axis.1, y = Axis.2, color = sampling_period)) + geom_point()

p1 <- plot_ordination(data_rare_noinsect_2023, ordination_pcoa, color="sampling_period") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p1

p2 <- plot_ordination(data_rare_noinsect_2023, ordination_pcoa, axes = c(1, 3), color="sampling_period") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p2

# Extract sample data
sample_data_df <- as.data.frame(sample_data(data_rare_noinsect_2023))

# Extract site scores from the ordination object
site_scores <- ordination_pcoa$vectors

# Combine scores with sample data
ordination_df <- data.frame(site_scores, sample_data_df)

# Plot the ordination using ggplot2
p <- ggplot(ordination_df, aes_string(x = "Axis.1", y = "Axis.2", color = "sampling_period")) +
  geom_point(size = 4) +
  stat_ellipse(aes(group = sampling_period), type = "norm") + # Add ellipses
  theme_minimal()
p

# Test whether the sampling periods ("Dates") differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_2023)$sampling_period)

# trying Jennifer's code below:
#bray.dist <- phyloseq::distance(ps.rarefied, method = "bray”)

#set.seed(53)

#adonis2(wunifrac_dist ~ sample_data(ps.rarefied)$Footprint*sample_data(ps.rarefied)$Subplot, by = #"margin”, permutations = #perm1) 
#If the interaction is not significant, you should rerun the model without the interaction. For this code, if the interaction is not significant, it won’t show you if the main effects are significant. 

# setting up the permutation object
perm1 <- with(sample_data(data_rare_noinsect_2023), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))
# dropping footprint in model
perm2 <- with(sample_data(data_rare_noinsect_2023), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_noinsect_2023)$sampling_period, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_2023)$sampling_period, by = "margin", permutations = perm2)
```


