---
title: "T7 warmx 16S Analysis 2021 & 2023"
author: "Moriah Young"
date: "2024-07-02; updated 2024-12-04"
output: pdf_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, echo=FALSE, message=FALSE}
rm(list = ls()) # clear working environment

# load packages
library(compositions)
library(cowplot)
library(cluster)
library(data.table)
library(devtools)
library(plyr)
library(dplyr)
library(doParallel)
library(factoextra)
library(ggplot2)
library(ggsignif)
library(gridExtra)
library(hillR)
library(microbiome)
library(multcomp)
library(nlme)
library(phyloseq)
library(qiime2R)
library(tidyverse)
library(vegan)
library(zCompositions)
library(permute)
library(pheatmap)
library(plotrix)
library(microViz)
#library(phylosmith)
#library(DESeq2)
#library(ANCOMBC)

# Set working directory from .Renviron
dir <- Sys.getenv("DATA_DIR")
list.files(dir)
```


```{r}
# 2021 data
# Create phyloseq objects of filtered data (16S)
phyloseq_2021 <- read_csv2phyloseq(otu.file = "/Users/moriahyoung/Desktop/phyloseq_object/16S-2021-merged-table-filtered.csv",
                                       taxonomy.file = "/Users/moriahyoung/Desktop/phyloseq_object/16S-2021-taxonomy-filtered.csv",
                                       metadata.file = "/Users/moriahyoung/Desktop/phyloseq_object/bacterial_2021_metadata.csv")
summarize_phyloseq(phyloseq_2021)

# 2023 data
# Create phyloseq objects of filtered data (16S)
phyloseq_2023 <- read_csv2phyloseq(otu.file = "/Users/moriahyoung/Desktop/phyloseq_object/16S-2023-table-filtered-noncontam.csv",
                                       taxonomy.file = "/Users/moriahyoung/Desktop/phyloseq_object/16S-2023-taxonomy-filtered-noncontam.csv",
                                       metadata.file = "/Users/moriahyoung/Desktop/phyloseq_object/bacterial_2023_metadata.csv")
summarize_phyloseq(phyloseq_2023)

# merge 2021 and 2023 phyloseq objects together
phyloseq_all <- merge_phyloseq(phyloseq_2021, phyloseq_2023)

samdat_tbl(phyloseq_all)
summarize_phyloseq(phyloseq_all)
```

#Filter data and rarefy data
```{r}
# filter data
# remove samples that have less than 1000 reads
data_16S_counts <- prune_samples(sample_sums(phyloseq_all)>=1000, phyloseq_all)
summarize_phyloseq(data_16S_counts)
# remove singletons
data_16S_counts1 <- filter_taxa(data_16S_counts, function (x) {sum(x > 0) > 1}, prune=TRUE)
summarize_phyloseq(data_16S_counts1)

# Rarefy data to min sample size reads 
set.seed(1022)
rare_16S <- rarefy_even_depth(data_16S_counts1, sample.size = min(sample_sums(data_16S_counts1)))
summarize_phyloseq(rare_16S)

# Need to change "Year" column to a factor
# Access the sample data
rare_16S_sample_data <- sample_data(rare_16S)

# Convert a specific column to a factor (e.g., "Year")
rare_16S_sample_data$Year <- as.factor(rare_16S_sample_data$Year)

# Update the phyloseq object with the modified sample data
sample_data(rare_16S) <- rare_16S_sample_data

# Check the structure to confirm
str(sample_data(rare_16S))

###########################################
# did not run code below
# Remove taxa not seen more than 3 times in at least 20% of the samples. This protects against an OTU with small mean & trivially large C.V.
phyloseq_all_filtered <- filter_taxa(phyloseq_all, function(x) sum(x > 3) > (0.2*length(x)), TRUE)
summarize_phyloseq(phyloseq_all_filtered)

# Rarefy data to min sample size reads 
set.seed(1022)
#rare_16S <- rarefy_even_depth(data_16S_counts, sample.size = 1000)
rare_16S <- rarefy_even_depth(phyloseq_all, sample.size = min(sample_sums(phyloseq_all)))
summarize_phyloseq(rare_16S)
sample_data(rare_16S)
```

# STACKED BAR PLOTS
Taxonomy
Setting up to make stacked bar plots
```{r}
data_16S_rel <- transform_sample_counts(rare_16S, function(x) x/sum(x))

# order sample data - "Drought" and "Subplot_Descriptions"
sample_data(data_16S_rel)$sampling_period <- ordered(sample_data(data_16S_rel)$sampling_period, c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"))
sample_data(data_16S_rel)$Subplot_Descriptions <- ordered(sample_data(data_16S_rel)$Subplot_Descriptions, c("irrigated_control", "ambient", "warmed", "warmed_insecticide", "insecticide", "drought", "warmed_drought", "drought_insecticide", "warmed_drought_insecticide"))

# subset to non insecticide plots
data_16S_rel_noinsect <- subset_samples(data_16S_rel, Subplot_Descriptions%in%c("ambient", "irrigated_control", "warmed_drought", "drought", "warmed"))
summarize_phyloseq(data_16S_rel_noinsect)

treatment_names <- c("ambient" = "Ambient",
                     "drought" = "Drought",
                     "irrigated_control" = "Irrigated \n Control",
                     "warmed" = "Warmed",
                     "warmed_drought" = "Warmed \n Drought", 
                     "drought_insecticide" = "Drought \n Insecticide", 
                     "insecticide" = "Insecticide", 
                     'warmed_drought_insecticide' = "Warmed \n Drought \n Insecticide", 
                     'warmed_insecticide' = "Warmed \n Insecticide")

treatment_names_noinsect <- c("ambient" = "Ambient",
                     "drought" = "Drought",
                     "irrigated_control" = "Irrigated \n Control",
                     "warmed" = "Warmed",
                     "warmed_drought" = "Warmed \n Drought")

# for faceting by treatment and year
treatment_names_year_noinsect <- c("ambient" = "Ambient",
                     "drought" = "Drought",
                     "irrigated_control" = "Irrigated \n Control",
                     "warmed" = "Warmed",
                     "warmed_drought" = "Warmed \n Drought",
                     "2021" = "2021",
                     "2023" = "2023")

custom_colors_15 <- c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1", "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4", "cyan2", "deeppink3", "goldenrod1", "aquamarine3", "darkviolet", "darkseagreen3")

custom_colors_10 <- c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1", "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")
```

# Phylum
Phylum Level - non insecticide plots
```{r}
phylum_sum_16S_noinsect <- tapply(taxa_sums(data_16S_rel_noinsect), 
                         tax_table(data_16S_rel_noinsect)[, "Phylum"], sum, na.rm=TRUE)
top10phylum_16S_noinsect <- names(sort(phylum_sum_16S_noinsect, TRUE))[1:15]

abundant_16S_phylum_noinsect <- subset_taxa(data_16S_rel_noinsect, Phylum %in% top10phylum_16S_noinsect)
abundant_16S_phylum_glom_noinsect <- tax_glom(abundant_16S_phylum_noinsect, taxrank = "Phylum")
abundant_16S_phylum_melt_noinsect <- psmelt(abundant_16S_phylum_glom_noinsect)

sum2_noinsect <- abundant_16S_phylum_melt_noinsect %>% 
  group_by(Subplot_Descriptions, sampling_period, Year, Phylum) %>%
  summarise(means = mean(Abundance))

#Everything that's left is considered rare  
rare <- sum2_noinsect %>%
  group_by(Subplot_Descriptions, sampling_period, Year) %>%
  summarise(means = 1- sum(means)) %>%
  mutate(Phylum = "Rare Taxa")

#concatenate the datasets
sum2_noinsect = rbind(sum2_noinsect, rare)

#order groups
sum2_noinsect$Phylum <- forcats::fct_relevel(sum2_noinsect$Phylum, "Rare Taxa", after = Inf)

# Stacked bar plot for top phylas
# facet by timing of soil collection (aka "Drought")
png("T7_warmx_ITS_order_stackedbarplot_by_samplingpoint.png", units="in", width=10, height=6, res=300)
ggplot(sum2_noinsect, aes(x = Subplot_Descriptions, y = means, fill = Phylum)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(Year~sampling_period, scale="free") +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
                                     "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4", "yellow", "gray20", "green", "purple", "pink")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
dev.off()

# facet by treatment

png("T7_warmx_16S_phylum_stackedbarplot_by_treatment_year_noinsect.png", units="in", width=8, height=6, res=300)
ggplot(sum2_noinsect, aes(x = sampling_period, y = means, fill = Phylum)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(Year~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names_year_noinsect)) +
        scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1", "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
         scale_x_discrete(labels=c("Pre-Drought" = "T1",
                                 "Peak-Drought" = "T2",
                                 "Post-Drought" = "T3",
                                 "Recovery" = "T4")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(size = 12),
              strip.text = element_text(size = 12, face = "italic"), 
              axis.title = element_blank(),
              title = element_text(size = 12),
              legend.position = "right", 
              legend.title = element_text(size = 12),
              legend.text = element_text(size = 12),
              axis.line = element_blank(),
              axis.text.y=element_text(size=12)) +
        theme(strip.text.y = element_text(size = 12, face = "italic", color = "black"))
dev.off()
```

# Order
Order Level - non insecticide plots
```{r}
order_sum_16S_noinsect <- tapply(taxa_sums(data_16S_rel_noinsect), 
                         tax_table(data_16S_rel_noinsect)[, "Order"], sum, na.rm=TRUE)
top10order_16S_noinsect <- names(sort(order_sum_16S_noinsect, TRUE))[1:15]

abundant_16S_order_noinsect <- subset_taxa(data_16S_rel_noinsect, Order %in% top10order_16S_noinsect)
abundant_16S_order_glom_noinsect <- tax_glom(abundant_16S_order_noinsect, taxrank = "Order")
abundant_16S_order_melt_noinsect <- psmelt(abundant_16S_order_glom_noinsect)

sum2_noinsect <- abundant_16S_order_melt_noinsect %>% 
  group_by(Subplot_Descriptions, sampling_period, Year, Order) %>%
  summarise(means = mean(Abundance))

#Everything that's left is considered rare  
rare <- sum2_noinsect %>%
  group_by(Subplot_Descriptions, sampling_period, Year) %>%
  summarise(means = 1- sum(means)) %>%
  mutate(Order = "Rare Taxa")

#concatenate the datasets
sum2_noinsect = rbind(sum2_noinsect, rare)

#order groups
sum2_noinsect$Order <- forcats::fct_relevel(sum2_noinsect$Order, "Rare Taxa", after = Inf)

# Stacked bar plot for top phylas
# facet by timing of soil collection (aka "Drought")
png("T7_warmx_ITS_order_stackedbarplot_by_samplingpoint.png", units="in", width=10, height=6, res=300)
ggplot(sum2_noinsect, aes(x = Subplot_Descriptions, y = means, fill = Order)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(Year~sampling_period, scale="free") +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        scale_fill_manual(values = custom_colors) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
dev.off()

# facet by treatment
png("T7_warmx_phylum_order_stackedbarplot_by_treatment_noinsect.png", units="in", width=8, height=8, res=300)
ggplot(sum2_noinsect, aes(x = sampling_period, y = means, fill = Order)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(Year~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names_year_noinsect)) +
        scale_fill_manual(values = custom_colors_15) +
         scale_x_discrete(labels=c("Pre-Drought" = "T1",
                                 "Peak-Drought" = "T2",
                                 "Post-Drought" = "T3",
                                 "Recovery" = "T4")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(size = 12),
              strip.text = element_text(size = 12, face = "italic"), 
              axis.title = element_blank(),
              title = element_text(size = 12),
              legend.position = "right", 
              legend.title = element_text(size = 12),
              legend.text = element_text(size = 12),
              axis.line = element_blank(),
              axis.text.y=element_text(size=12))
dev.off()

```

# Testing rel abun differences
```{r}
# Aggregate Data at the order Level
physeq_order <- tax_glom(rare_16S, taxrank = "Order")

# transform to relative abundance
physeq_order_rel <- transform_sample_counts(physeq_order, function(x) x / sum(x))

physeq_order_rel_noinsect <- subset_samples(physeq_order_rel, Subplot_Descriptions%in%c("ambient", "irrigated_control", "warmed_drought", "drought", "warmed"))

# Convert the phyloseq object into a data frame for statistical analysis
physeq_order_df <- psmelt(physeq_order_rel_noinsect)

# are there differences in rel abun based on sampling period?
kruskal.test(Abundance ~ sampling_period, data = physeq_order_df)
# Kruskal-Wallis chi-squared = 19.991, df = 3, p-value = 0.0001704

kruskal.test(Abundance ~ Subplot_Descriptions, data = physeq_order_df)
# Kruskal-Wallis chi-squared = 6.4644, df = 4, p-value = 0.167

kruskal.test(Abundance ~ Year, data = physeq_order_df)
# Kruskal-Wallis chi-squared = 11.047, df = 1, p-value = 0.0008883

# Filter for Specific Phylum (Optional)
physeq_df_filtered <- subset(physeq_order_df, Order == "Pezizales")

kruskal.test(Abundance ~ sampling_period, data = physeq_df_filtered)
# KKruskal-Wallis chi-squared = 4.8517, df = 3, p-value = 0.183
kruskal.test(Abundance ~ Subplot_Descriptions, data = physeq_df_filtered)
# Kruskal-Wallis chi-squared = 3.711, df = 4, p-value = 0.4465

physeq_df_filtered_1 <- subset(physeq_order_df, Order == "Agaricales")
kruskal.test(Abundance ~ sampling_period, data = physeq_df_filtered_1) # NS
kruskal.test(Abundance ~ Drought, data = physeq_df_filtered_1) # NS

# test sampling periods within specific treatments
physeq_df_filtered_wd <- subset(physeq_order_df, Subplot_Descriptions == "warmed_drought")
kruskal.test(Abundance ~ sampling_period, data = physeq_df_filtered_wd)
```

# Family
Family Level - non insecticide plots
```{r}
family_sum_16S_noinsect <- tapply(taxa_sums(data_16S_rel_noinsect), 
                         tax_table(data_16S_rel_noinsect)[, "Family"], sum, na.rm=TRUE)
top10family_16S_noinsect <- names(sort(family_sum_16S_noinsect, TRUE))[1:15]

abundant_16S_family_noinsect <- subset_taxa(data_16S_rel_noinsect, Family %in% top10family_16S_noinsect)
abundant_16S_family_glom_noinsect <- tax_glom(abundant_16S_family_noinsect, taxrank = "Family")
abundant_16S_family_melt_noinsect <- psmelt(abundant_16S_family_glom_noinsect)

sum2_noinsect <- abundant_16S_family_melt_noinsect %>% 
  group_by(Subplot_Descriptions, sampling_period, Year, Family) %>%
  summarise(means = mean(Abundance))

#Everything that's left is considered rare  
rare <- sum2_noinsect %>%
  group_by(Subplot_Descriptions, sampling_period, Year) %>%
  summarise(means = 1- sum(means)) %>%
  mutate(Family = "Rare Taxa")

#concatenate the datasets
sum2_noinsect = rbind(sum2_noinsect, rare)

#order groups
sum2_noinsect$Family <- forcats::fct_relevel(sum2_noinsect$Family, "Rare Taxa", after = Inf)

# Stacked bar plot for top phylas
# facet by timing of soil collection (aka "Drought")
png("T7_warmx_ITS_order_stackedbarplot_by_samplingpoint.png", units="in", width=10, height=6, res=300)
ggplot(sum2_noinsect, aes(x = Subplot_Descriptions, y = means, fill = Family)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(Year~sampling_period, scale="free") +
        #theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        #scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
#                                     "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
dev.off()

# facet by treatment
png("T7_warmx_ITS_order_stackedbarplot_by_treatment_noinsect.png", units="in", width=8, height=6, res=300)
ggplot(sum2_noinsect, aes(x = sampling_period, y = means, fill = Family)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(Year~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names_noinsect)) +
        scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1", "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
         scale_x_discrete(labels=c("Pre-Drought" = "T1",
                                 "Peak-Drought" = "T2",
                                 "Post-Drought" = "T3",
                                 "Recovery" = "T4")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(size = 12),
              strip.text = element_text(size = 12, face = "italic"), 
              axis.title = element_blank(),
              title = element_text(size = 12),
              legend.position = "right", 
              legend.title = element_text(size = 12),
              legend.text = element_text(size = 12),
              axis.line = element_blank(),
              axis.text.y=element_text(size=12))
dev.off()
```

# Genus
Genus Level - non insecticide plots
```{r}
genus_sum_16S_noinsect <- tapply(taxa_sums(data_16S_rel_noinsect), 
                         tax_table(data_16S_rel_noinsect)[, "Genus"], sum, na.rm=TRUE)
top10genus_16S_noinsect <- names(sort(genus_sum_16S_noinsect, TRUE))[1:10]

abundant_16S_genus_noinsect <- subset_taxa(data_16S_rel_noinsect, Genus %in% top10genus_16S_noinsect)
abundant_16S_genus_glom_noinsect <- tax_glom(abundant_16S_genus_noinsect, taxrank = "Genus")
abundant_16S_genus_melt_noinsect <- psmelt(abundant_16S_genus_glom_noinsect)

sum2_noinsect <- abundant_16S_genus_melt_noinsect %>% 
  group_by(Subplot_Descriptions, sampling_period, Year, Genus) %>%
  summarise(means = mean(Abundance))

#Everything that's left is considered rare  
rare <- sum2_noinsect %>%
  group_by(Subplot_Descriptions, sampling_period, Year) %>%
  summarise(means = 1- sum(means)) %>%
  mutate(Genus= "Rare Taxa")

#concatenate the datasets
sum2_noinsect = rbind(sum2_noinsect, rare)

#order groups
sum2_noinsect$Genus <- forcats::fct_relevel(sum2_noinsect$Genus, "Rare Taxa", after = Inf)

# Stacked bar plot for top phylas
# facet by timing of soil collection (aka "Drought")
png("T7_warmx_ITS_order_stackedbarplot_by_samplingpoint.png", units="in", width=10, height=6, res=300)
ggplot(sum2_noinsect, aes(x = Subplot_Descriptions, y = means, fill = Genus)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(Year~sampling_period, scale="free") +
        theme(legend.position = "right", legend.title = element_blank(), axis.line = element_blank()) +
        scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1",
                                     "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
        scale_x_discrete(labels=c("ambient" = "Ambient",
                                  "drought" = "Drought",
                                  "irrigated_control" = "Irrigated Control",
                                  "warmed" = "Warmed",
                                  "warmed_drought" = "Warmed Drought")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(angle = 60, hjust = 1)) +
        theme(strip.text = element_text(size = 12, face = "italic"), axis.text.x = element_text(size = 10),
              axis.title = element_blank(),
              title = element_text(size = 12))
dev.off()

# facet by treatment
png("T7_warmx_ITS_order_stackedbarplot_by_treatment_noinsect.png", units="in", width=8, height=6, res=300)
ggplot(sum2_noinsect, aes(x = sampling_period, y = means, fill = Genus)) +
        geom_bar(position = "stack", stat = "identity", col = "black") +
        facet_grid(Year~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names_noinsect)) +
        scale_fill_manual(values = c("firebrick4","tomato", "darkorange3", "orange1" ,"goldenrod4", "khaki1", "forestgreen", "olivedrab1", "midnightblue", "powderblue", "mediumorchid4")) +
         scale_x_discrete(labels=c("Pre-Drought" = "T1",
                                 "Peak-Drought" = "T2",
                                 "Post-Drought" = "T3",
                                 "Recovery" = "T4")) +
        #scale_y_continuous(name = NULL, breaks = NULL) +
        panel_border() +
        theme_bw() +
        ggtitle("") +
        theme(axis.text.x = element_text(size = 12),
              strip.text = element_text(size = 12, face = "italic"), 
              axis.title = element_blank(),
              title = element_text(size = 12),
              legend.position = "right", 
              legend.title = element_text(size = 12),
              legend.text = element_text(size = 12),
              axis.line = element_blank(),
              axis.text.y=element_text(size=12))
dev.off()
```

# 2023 
```{r}
# subset for 2023 data
data_rare_2023 <- subset_samples(data_rare_noinsect, Year%in%c("2023"))
data_rare_2023 <- subset_samples(data_rare_2023, !is.na(sampling_period))
data_rare_2023_sample <- sample_data(data_rare_2023)

# PCoA plot using bray-curtis as distance
bray_dist <- phyloseq::distance(data_rare_2023, method="bray", weighted=T)
ordination <- ordinate(data_rare_2023, method="PCoA", distance=bray_dist)

png("16S_pcoa_no_insects.png", units="in", width=8, height=6, res=300)
plot_ordination(data_rare_2023, ordination) + 
        geom_point(aes(color = Subplot_Descriptions, shape = sampling_period), size=4, alpha=0.75) +
        stat_ellipse(aes(color = Subplot_Descriptions), geom = "polygon", alpha = .1) +
            scale_colour_manual(name = "Subplot Descriptions", 
                          breaks = c("irrigated_control", "ambient", "warmed", "drought", "warmed_drought"),
                          labels = c("Irrigated Control", "Ambient", "Warmed", "Drought", "Warmed + Drought"),
                          values = c("ambient" = "#a6bddb", "drought" = "#687689", "warmed" = "#fb6a4a", "warmed_drought" = "#9D422E")) +
        scale_shape_manual(name = "Sampling Time", 
                       breaks = c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"),
                       labels = c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"),
                       values = c(16, 17, 15, 18)) +
        theme(aspect.ratio=1)
dev.off()

# Test whether the sampling periods differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_2023)$Subplot_Descriptions)
adonis2(bray_dist ~ sample_data(data_rare_2023)$sampling_period)

# setting up the permutation object
perm1 <- with(sample_data(data_rare_2023), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))

perm2 <- with(sample_data(data_rare_2023), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_2023)$Year, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_2023)$sampling_period, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_2023)$Subplot_Descriptions*sample_data(data_rare_2023)$sampling_period, by = "margin", permutations = perm1) # interaction not significatn
adonis2(bray_dist ~ sample_data(data_rare_2023)$Subplot_Descriptions + sample_data(data_rare_2023)$sampling_period, by = "margin", permutations = perm1)
```


# PEAK DROUGHT NO IRRIGATED CONTROL
#Subsetting phyloseq object
```{r}
# no insecticide treatment and no irrigated control
data_rare_noinsect <- subset_samples(rare_16S, Subplot_Descriptions%in%c("ambient", "warmed_drought", "drought", "warmed"))

# peak drought only
data_rare_noinsect_nona <- subset_samples(data_rare_noinsect, !is.na(sampling_period))
data_rare_noinsect_peak <- subset_samples(data_rare_noinsect_nona, sampling_period == "Peak-Drought")
summarize_phyloseq(data_rare_noinsect_peak)
```

#Alpha Diversity
```{r}
# order sample data - "sampling_period" and "Subplot_Descriptions"
sample_data(data_rare_noinsect_peak)$sampling_period <- ordered(sample_data(data_rare_noinsect_peak)$sampling_period, c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"))
sample_data(data_rare_noinsect_peak)$Subplot_Descriptions <- ordered(sample_data(data_rare_noinsect_peak)$Subplot_Descriptions, c( "ambient", "warmed","drought", "warmed_drought"))

# Plotting Alpha Diversity
treatment_names <- c("ambient" = "Ambient",
                     "drought" = "Drought",
                     "warmed" = "Warmed",
                     "warmed_drought" = "Warmed + Drought")

plot_richness(data_rare_noinsect_peak, x="Year", color="Subplot_Descriptions", measures=c("Observed"))
plot_richness(data_rare_noinsect_peak, x="Year", measures=c("Observed", "Shannon")) + geom_boxplot()
plot_richness(data_rare_noinsect_peak, x="Subplot_Descriptions", measures=c("Observed")) + geom_boxplot() + facet_grid(.~Year)
plot_richness(data_rare_noinsect_peak, x="Subplot_Descriptions", measures=c("Shannon")) + geom_boxplot() + facet_grid(.~Year)


png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_peak, x="Year", measures=c("Observed")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(y = "Observed OTUs", x = "Time of Sampling") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
dev.off()

#png("T7_warmx_16S_shannon_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_peak, x="Year", measures=c("Shannon")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(x = "Time of Sampling", y = "Shannon Index") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()

# analysis of alpha diversity
richness <- estimate_richness(data_rare_noinsect_peak)
head(richness)
kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_peak)$Subplot_Descriptions)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_peak)$Subplot_Descriptions, p.adj = "bonf")

kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_peak)$Year)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_peak)$Year, p.adj = "bonf")
```

Beta Diversity
```{r}
# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(data_rare_noinsect_peak, method="bray")
ordination_pcoa = ordinate(data_rare_noinsect_peak, method="PCoA", distance=bray_dist)

# https://stackoverflow.com/questions/76465183/plot-ordination-function-of-phyloseq-plots-shapes-double-within-itself
# Save the PCoA vectors data from your ordinated object
vectors_peak <- as.data.frame(ordination_pcoa$vectors)
values_peak <- as.data.frame(ordination_pcoa$values)

library(tibble)
vectors_peak <- rownames_to_column(vectors_peak, "SampleID")

metadata_peak <- rownames_to_column(data.frame(data_rare_noinsect_peak@sam_data), "SampleID")
vectors_metadata_peak <- dplyr::inner_join(vectors_peak, metadata_peak, by = "SampleID")

ggplot(vectors_metadata_peak, aes(x = Axis.1, y = Axis.2, color = Year)) + geom_point()

p1 <- plot_ordination(data_rare_noinsect_peak, ordination_pcoa, color="Year", shape = "Subplot_Descriptions") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Subplot_Descriptions), geom = "polygon", alpha = .1) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p1

p2 <- plot_ordination(data_rare_noinsect_peak, ordination_pcoa, axes = c(1, 3), color="Year") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p2

# Extract sample data
sample_data_df <- as.data.frame(sample_data(data_rare_noinsect_peak))

# Extract site scores from the ordination object
site_scores <- ordination_pcoa$vectors

# Combine scores with sample data
ordination_df <- data.frame(site_scores, sample_data_df)

# Plot the ordination using ggplot2
p <- ggplot(ordination_df, aes_string(x = "Axis.1", y = "Axis.2", color = "Year", shape = "Subplot_Descriptions")) +
  geom_point(size = 4) +
  stat_ellipse(aes(group = Subplot_Descriptions), type = "norm") + # Add ellipses
  theme_minimal()
p

# Test whether the sampling periods differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_peak)$Year)

# trying Jennifer's code below:
#bray.dist <- phyloseq::distance(ps.rarefied, method = "bray”)

#set.seed(53)

#adonis2(wunifrac_dist ~ sample_data(ps.rarefied)$Footprint*sample_data(ps.rarefied)$Subplot, by = #"margin”, permutations = #perm1) 
#If the interaction is not significant, you should rerun the model without the interaction. For this code, if the interaction is not significant, it won’t show you if the main effects are significant. 

# setting up the permutation object
perm1 <- with(sample_data(data_rare_noinsect_peak), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))

perm2 <- with(sample_data(data_rare_noinsect_peak), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_noinsect_peak)$Year, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_peak)$Year, by = "margin", permutations = perm2)
```

# POST DROUGHT NO IRRIGATED CONTROL
#Subsetting phyloseq object
```{r}
# post drought only
data_rare_noinsect_post <- subset_samples(data_rare_noinsect_nona, sampling_period == "Post-Drought")
summarize_phyloseq(data_rare_noinsect_post)
```

#Alpha Diversity
```{r}
# order sample data - "Subplot_Descriptions"
sample_data(data_rare_noinsect_post)$Subplot_Descriptions <- ordered(sample_data(data_rare_noinsect_post)$Subplot_Descriptions, c( "ambient", "warmed","drought", "warmed_drought"))

plot_richness(data_rare_noinsect_post, x="Year", color="Subplot_Descriptions", measures=c("Observed"))
plot_richness(data_rare_noinsect_post, x="Year", measures=c("Observed", "Shannon")) + geom_boxplot()
plot_richness(data_rare_noinsect_post, x="Subplot_Descriptions", measures=c("Observed")) + geom_boxplot() + facet_grid(.~Year)
plot_richness(data_rare_noinsect_peak, x="Subplot_Descriptions", measures=c("Shannon")) + geom_boxplot() + facet_grid(.~Year)


#png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_post, x="Year", measures=c("Observed")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(y = "Observed OTUs", x = "Year") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
#dev.off()

#png("T7_warmx_16S_shannon_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_post, x="Year", measures=c("Shannon")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(x = "Year", y = "Shannon Index") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()

# analysis of alpha diversity
richness <- estimate_richness(data_rare_noinsect_post)
head(richness)
kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_post)$Subplot_Descriptions)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_post)$Subplot_Descriptions, p.adj = "bonf")

kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_post)$Year)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_post)$Year, p.adj = "bonf")
```

Beta Diversity
```{r}
# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(data_rare_noinsect_post, method="bray")
ordination_pcoa = ordinate(data_rare_noinsect_post, method="PCoA", distance=bray_dist)

# https://stackoverflow.com/questions/76465183/plot-ordination-function-of-phyloseq-plots-shapes-double-within-itself
# Save the PCoA vectors data from your ordinated object
vectors_post <- as.data.frame(ordination_pcoa$vectors)
values_post <- as.data.frame(ordination_pcoa$values)

library(tibble)
vectors_post <- rownames_to_column(vectors_post, "SampleID")

metadata_post <- rownames_to_column(data.frame(data_rare_noinsect_post@sam_data), "SampleID")
vectors_metadata_post <- dplyr::inner_join(vectors_post, metadata_post, by = "SampleID")

ggplot(vectors_metadata_post, aes(x = Axis.1, y = Axis.2, color = Year)) + geom_point()

p1 <- plot_ordination(data_rare_noinsect_post, ordination_pcoa, color="Year") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p1

p2 <- plot_ordination(data_rare_noinsect_post, ordination_pcoa, axes = c(1, 3), color="Year") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p2

# Extract sample data
sample_data_df <- as.data.frame(sample_data(data_rare_noinsect_post))

# Extract site scores from the ordination object
site_scores <- ordination_pcoa$vectors

# Combine scores with sample data
ordination_df <- data.frame(site_scores, sample_data_df)

# Plot the ordination using ggplot2
p <- ggplot(ordination_df, aes_string(x = "Axis.1", y = "Axis.2", color = "Year")) +
  geom_point(size = 4) +
  stat_ellipse(aes(group = Year), type = "norm") + # Add ellipses
  theme_minimal()
p

# Test whether the sampling periods ("Dates") differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_post)$Year)

# trying Jennifer's code below:
#bray.dist <- phyloseq::distance(ps.rarefied, method = "bray”)

#set.seed(53)

#adonis2(wunifrac_dist ~ sample_data(ps.rarefied)$Footprint*sample_data(ps.rarefied)$Subplot, by = #"margin”, permutations = #perm1) 
#If the interaction is not significant, you should rerun the model without the interaction. For this code, if the interaction is not significant, it won’t show you if the main effects are significant. 

# setting up the permutation object
perm1 <- with(sample_data(data_rare_noinsect_post), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))
# dropping footprint in model
perm2 <- with(sample_data(data_rare_noinsect_post), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_noinsect_post)$Year, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_post)$Year, by = "margin", permutations = perm2)
```

# 2023
#Subsetting phyloseq object
```{r}
# post drought only
data_rare_noinsect_2023 <- subset_samples(data_rare_noinsect_nona, Year == "2023")
summarize_phyloseq(data_rare_noinsect_2023)
```

#Alpha diversity
```{r}
# order sample data - "Subplot_Descriptions"
sample_data(data_rare_noinsect_2023)$Subplot_Descriptions <- ordered(sample_data(data_rare_noinsect_2023)$Subplot_Descriptions, c( "ambient", "warmed","drought", "warmed_drought"))
# "sampling_period"
sample_data(data_rare_noinsect_2023)$sampling_period <- ordered(sample_data(data_rare_noinsect_2023)$sampling_period, c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"))

plot_richness(data_rare_noinsect_2023, x="sampling_period", color="Subplot_Descriptions", measures=c("Observed"))
plot_richness(data_rare_noinsect_2023, x="sampling_period", measures=c("Observed", "Shannon")) + geom_boxplot()
plot_richness(data_rare_noinsect_2023, x="Subplot_Descriptions", measures=c("Observed")) + geom_boxplot() + facet_grid(.~sampling_period)
plot_richness(data_rare_noinsect_2023, x="Subplot_Descriptions", measures=c("Shannon")) + geom_boxplot() + facet_grid(.~sampling_period)


#png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_2023, x="sampling_period", measures=c("Observed")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(y = "Observed OTUs", x = "Sampling Period") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
#dev.off()

#png("T7_warmx_16S_shannon_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_2023, x="sampling_period", measures=c("Shannon")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(x = "Year", y = "Shannon Index") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        scale_fill_manual(values = c("ambient" = "darkblue", "drought" = "gray58", 
        "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()

# analysis of alpha diversity
richness <- estimate_richness(data_rare_noinsect_2023)
head(richness)
kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_2023)$Subplot_Descriptions)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_post)$Subplot_Descriptions, p.adj = "bonf")

kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_2023)$sampling_period)
pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_post)$sampling_period, p.adj = "bonf")
```

Beta Diversity
```{r}
# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(data_rare_noinsect_2023, method="bray")
ordination_pcoa = ordinate(data_rare_noinsect_2023, method="PCoA", distance=bray_dist)

# https://stackoverflow.com/questions/76465183/plot-ordination-function-of-phyloseq-plots-shapes-double-within-itself
# Save the PCoA vectors data from your ordinated object
vectors_2023 <- as.data.frame(ordination_pcoa$vectors)
values_2023 <- as.data.frame(ordination_pcoa$values)

library(tibble)
vectors_2023 <- rownames_to_column(vectors_2023, "SampleID")

metadata_2023 <- rownames_to_column(data.frame(data_rare_noinsect_2023@sam_data), "SampleID")
vectors_metadata_2023 <- dplyr::inner_join(vectors_2023, metadata_2023, by = "SampleID")

ggplot(vectors_metadata_2023, aes(x = Axis.1, y = Axis.2, color = sampling_period)) + geom_point()

p1 <- plot_ordination(data_rare_noinsect_2023, ordination_pcoa, color="sampling_period") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p1

p2 <- plot_ordination(data_rare_noinsect_2023, ordination_pcoa, axes = c(1, 3), color="sampling_period") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p2

# Extract sample data
sample_data_df <- as.data.frame(sample_data(data_rare_noinsect_2023))

# Extract site scores from the ordination object
site_scores <- ordination_pcoa$vectors

# Combine scores with sample data
ordination_df <- data.frame(site_scores, sample_data_df)

# Plot the ordination using ggplot2
p <- ggplot(ordination_df, aes_string(x = "Axis.1", y = "Axis.2", color = "sampling_period")) +
  geom_point(size = 4) +
  stat_ellipse(aes(group = sampling_period), type = "norm") + # Add ellipses
  theme_minimal()
p

# Test whether the sampling periods ("Dates") differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_2023)$sampling_period)

# trying Jennifer's code below:
#bray.dist <- phyloseq::distance(ps.rarefied, method = "bray”)

#set.seed(53)

#adonis2(wunifrac_dist ~ sample_data(ps.rarefied)$Footprint*sample_data(ps.rarefied)$Subplot, by = #"margin”, permutations = #perm1) 
#If the interaction is not significant, you should rerun the model without the interaction. For this code, if the interaction is not significant, it won’t show you if the main effects are significant. 

# setting up the permutation object
perm1 <- with(sample_data(data_rare_noinsect_2023), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))
# dropping footprint in model
perm2 <- with(sample_data(data_rare_noinsect_2023), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_noinsect_2023)$sampling_period, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_2023)$sampling_period, by = "margin", permutations = perm2)
```

# USE FOR KBS LTER FELLOWS PRESENTATION
DROUGHT @ peak drought 2021 & 2023
Subsetting phyloseq object
```{r}
data_rare_noinsect_drought <- subset_samples(data_rare_noinsect_nona, Subplot_Descriptions == "drought")
summarize_phyloseq(data_rare_noinsect_drought)

data_rare_noinsect_drought_peak <- subset_samples(data_rare_noinsect_drought, sampling_period == "Peak-Drought")
```

#Alpha Diversity
```{r}
# order sample data - "sampling_period" and "Subplot_Descriptions"
sample_data(data_rare_noinsect_drought_peak)$sampling_period <- ordered(sample_data(data_rare_noinsect_drought_peak)$sampling_period, c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"))
sample_data(data_rare_noinsect_drought_peak)$Subplot_Descriptions <- ordered(sample_data(data_rare_noinsect_drought_peak)$Subplot_Descriptions, c( "ambient", "warmed","drought", "warmed_drought"))

plot_richness(data_rare_noinsect_drought_peak, x="Year", color="Subplot_Descriptions", measures=c("Observed"))
plot_richness(data_rare_noinsect_drought_peak, x="Year", measures=c("Observed", "Shannon")) + geom_boxplot()

png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_drought_peak, x="Year", measures=c("Observed")) + 
        geom_boxplot(aes(fill = Subplot_Descriptions)) + 
        labs(y = "Observed OTUs", x = "Time of Sampling") +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
dev.off()

#png("T7_warmx_16S_shannon_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_drought_peak, x="Year", measures=c("Shannon")) + 
        geom_boxplot(aes(fill = Year)) + 
        labs(x = "Time of Sampling", y = "Shannon Index") +
        facet_grid(.~Subplot_Descriptions, scale="free", labeller = as_labeller(treatment_names)) +
        theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()

# analysis of alpha diversity
richness <- estimate_richness(data_rare_noinsect_drought_peak)
head(richness)
kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_drought_peak)$Year)
#pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_drought_peak)$Year, p.adj = "bonf")
```




#Beta diversity
```{r}
# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(data_rare_noinsect_drought_peak, method="bray")
ordination_pcoa = ordinate(data_rare_noinsect_drought_peak, method="PCoA", distance=bray_dist)

# https://stackoverflow.com/questions/76465183/plot-ordination-function-of-phyloseq-plots-shapes-double-within-itself
# Save the PCoA vectors data from your ordinated object
vectors_peak <- as.data.frame(ordination_pcoa$vectors)
values_peak <- as.data.frame(ordination_pcoa$values)

library(tibble)
vectors_peak <- rownames_to_column(vectors_peak, "SampleID")

metadata_peak <- rownames_to_column(data.frame(data_rare_noinsect_peak@sam_data), "SampleID")
vectors_metadata_peak <- dplyr::inner_join(vectors_peak, metadata_peak, by = "SampleID")

ggplot(vectors_metadata_peak, aes(x = Axis.1, y = Axis.2, color = Year)) + geom_point()

p1 <- plot_ordination(data_rare_noinsect_drought_peak, ordination_pcoa, color="Year") + 
        geom_point(size=5, alpha=0.90) +
        stat_ellipse(aes(fill = Year), geom = "polygon", alpha = .1) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
png("16S_peak_drought_2021_2023.png", units="in", width=8, height=6, res=300)
p1

dev.off()

p2 <- plot_ordination(data_rare_noinsect_peak, ordination_pcoa, axes = c(1, 3), color="Year") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p2

# Extract sample data
sample_data_df <- as.data.frame(sample_data(data_rare_noinsect_drought_peak))

# Extract site scores from the ordination object
site_scores <- ordination_pcoa$vectors

# Combine scores with sample data
ordination_df <- data.frame(site_scores, sample_data_df)

# Plot the ordination using ggplot2
p <- ggplot(ordination_df, aes_string(x = "Axis.1", y = "Axis.2", color = "Year", shape = "Subplot_Descriptions")) +
  geom_point(size = 4) +
  stat_ellipse(aes(group = Year), type = "norm") + # Add ellipses
  theme_minimal()
p

# Test whether the sampling periods differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_drought_peak)$Year)

# setting up the permutation object
perm1 <- with(sample_data(data_rare_noinsect_drought_peak), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))

perm2 <- with(sample_data(data_rare_noinsect_drought_peak), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_noinsect_drought_peak)$Year, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_drought_peak)$Year, by = "margin", permutations = perm2)
```

# NMDS
```{r}
# bray curtis ordination
nmds_bray_16S <- ordinate(
        physeq = ord_16S, 
        method = "NMDS", 
        distance = "bray"
        )

# plot ordination
plot_ordination(ord_16S, nmds_bray_16S, "samples", color = "Subplot_Descriptions")
# Run 20 stress 0.137421
#png("T7_warmx_16S_braycurtis_NMDS_ordination.png", units="in", width=10, height=6, res=300)
plot_ordination(
  physeq = ord_16S,                                                         #phyloseq object
  ordination = bray_pcoa_16S) +                                                #ordination
  geom_point(aes(fill = Subplot_Descriptions, shape = Drought), size = 3) + #sets fill color to sampletype
        stat_ellipse(aes(fill = factor(Subplot_Descriptions)), geom = "polygon", alpha = .5) +
        scale_fill_manual(name = "Subplot Descriptions", 
                          breaks = c("irrigated_control", "ambient", "warmed", "drought", "warmed_drought"),
                          labels = c("Irrigated Control", "Ambient", "Warmed", "Drought", "Warmed + Drought"),
                          values = c("ambient" = "darkblue", "drought" = "gray58", "irrigated_control" = "deepskyblue", "warmed" = "red2", "warmed_drought" = "darkred")) +
        theme(legend.title = element_blank())                                      #removes legend title
#dev.off()
```

# DROUGHT 2021 & 2023
#Subsetting phyloseq object
```{r}
data_rare_noinsect_drought <- subset_samples(data_rare_noinsect_nona, Subplot_Descriptions == "drought")
summarize_phyloseq(data_rare_noinsect_drought)
```

#Alpha Diversity
```{r}
# order sample data - "sampling_period" and "Subplot_Descriptions"
sample_data(data_rare_noinsect_drought)$sampling_period <- ordered(sample_data(data_rare_noinsect_drought)$sampling_period, c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"))

plot_richness(data_rare_noinsect_drought, x="Year", color="Year", measures=c("Observed"))
plot_richness(data_rare_noinsect_drought, x="Year", measures=c("Observed", "Shannon")) + geom_boxplot()
plot_richness(data_rare_noinsect_drought, x="sampling_period", color="Year", measures=c("Observed"))
plot_richness(data_rare_noinsect_drought, x="sampling_period", measures=c("Observed", "Shannon")) + geom_boxplot()

#png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_drought, x="sampling_period", measures=c("Observed")) + 
        geom_boxplot(aes(fill = sampling_period)) + 
        labs(y = "Observed OTUs", x = "Time of Sampling") +
        facet_grid(.~Year, scale="free") +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
#dev.off()

#png("T7_warmx_16S_shannon_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_drought, x="sampling_period", measures=c("Shannon")) + 
        geom_boxplot(aes(fill = sampling_period)) + 
        labs(x = "Time of Sampling", y = "Shannon Index") +
        facet_grid(.~Year, scale="free") +
        theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()

# analysis of alpha diversity
richness <- estimate_richness(data_rare_noinsect_drought)
head(richness)
kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_drought)$Year)
#pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_drought_peak)$Year, p.adj = "bonf")
```

#Beta Diversity
```{r}
# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(data_rare_noinsect_drought, method="bray")
ordination_pcoa = ordinate(data_rare_noinsect_drought, method="PCoA", distance=bray_dist)

# https://stackoverflow.com/questions/76465183/plot-ordination-function-of-phyloseq-plots-shapes-double-within-itself
# Save the PCoA vectors data from your ordinated object
vectors_peak <- as.data.frame(ordination_pcoa$vectors)
values_peak <- as.data.frame(ordination_pcoa$values)

library(tibble)
vectors_peak <- rownames_to_column(vectors_peak, "SampleID")

metadata_peak <- rownames_to_column(data.frame(data_rare_noinsect_peak@sam_data), "SampleID")
vectors_metadata_peak <- dplyr::inner_join(vectors_peak, metadata_peak, by = "SampleID")

ggplot(vectors_metadata_peak, aes(x = Axis.1, y = Axis.2, color = Year)) + geom_point()

p1 <- plot_ordination(data_rare_noinsect_drought, ordination_pcoa, color="Year") + 
  geom_point(size=5, alpha=0.90) +
  stat_ellipse(aes(fill = Year), geom = "polygon", alpha = .1) + 
  theme(aspect.ratio=1) +
  scale_color_manual(values = c("2021" = "gray58", "2023" = "gray20")) +  # Set point colors
  scale_fill_manual(values = c("2021" = "gray58", "2023" = "gray20"))     # Set ellipse fill colors
png("16S_drought_2021_2023.png", units="in", width=8, height=6, res=300)
p1
dev.off()

p2 <- plot_ordination(data_rare_noinsect_drought, ordination_pcoa, axes = c(1, 3), color="Year") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p2

# Extract sample data
sample_data_df <- as.data.frame(sample_data(data_rare_noinsect_drought_peak))

# Extract site scores from the ordination object
site_scores <- ordination_pcoa$vectors

# Combine scores with sample data
ordination_df <- data.frame(site_scores, sample_data_df)

# Plot the ordination using ggplot2
p <- ggplot(ordination_df, aes_string(x = "Axis.1", y = "Axis.2", color = "Year", shape = "Subplot_Descriptions")) +
  geom_point(size = 4) +
  stat_ellipse(aes(group = Year), type = "norm") + # Add ellipses
  theme_minimal()
p

# Test whether the sampling periods differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_drought)$Year)

# setting up the permutation object
perm1 <- with(sample_data(data_rare_noinsect_drought), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))

perm2 <- with(sample_data(data_rare_noinsect_drought), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_noinsect_drought)$Year, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_drought)$sampling_period, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_drought)$Year*sample_data(data_rare_noinsect_drought)$sampling_period, by = "margin", permutations = perm1)
```

# WARMED + DROUGHT 2021 & 2023
#Subsetting phyloseq object
```{r}
data_rare_noinsect_wd <- subset_samples(data_rare_noinsect_nona, Subplot_Descriptions == "warmed_drought")
summarize_phyloseq(data_rare_noinsect_wd)
```

#Alpha Diversity
```{r}
# order sample data - "sampling_period" and "Subplot_Descriptions"
sample_data(data_rare_noinsect_wd)$sampling_period <- ordered(sample_data(data_rare_noinsect_wd)$sampling_period, c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"))

plot_richness(data_rare_noinsect_wd, x="Year", color="Year", measures=c("Observed"))
plot_richness(data_rare_noinsect_wd, x="Year", measures=c("Observed", "Shannon")) + geom_boxplot()
plot_richness(data_rare_noinsect_wd, x="sampling_period", color="Year", measures=c("Observed"))
plot_richness(data_rare_noinsect_wd, x="sampling_period", measures=c("Observed", "Shannon")) + geom_boxplot()

#png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_wd, x="sampling_period", measures=c("Observed")) + 
        geom_boxplot(aes(fill = sampling_period)) + 
        labs(y = "Observed OTUs", x = "Time of Sampling") +
        facet_grid(.~Year, scale="free") +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
#dev.off()

#png("T7_warmx_16S_shannon_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_wd, x="sampling_period", measures=c("Shannon")) + 
        geom_boxplot(aes(fill = sampling_period)) + 
        labs(x = "Time of Sampling", y = "Shannon Index") +
        facet_grid(.~Year, scale="free") +
        theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()

# analysis of alpha diversity
richness <- estimate_richness(data_rare_noinsect_wd)
head(richness)
kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_wd)$Year)
#pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_drought_peak)$Year, p.adj = "bonf")
```

#Beta Diversity
```{r}
# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(data_rare_noinsect_wd, method="bray")
ordination_pcoa = ordinate(data_rare_noinsect_wd, method="PCoA", distance=bray_dist)

# https://stackoverflow.com/questions/76465183/plot-ordination-function-of-phyloseq-plots-shapes-double-within-itself
# Save the PCoA vectors data from your ordinated object
vectors_peak <- as.data.frame(ordination_pcoa$vectors)
values_peak <- as.data.frame(ordination_pcoa$values)

library(tibble)
vectors_peak <- rownames_to_column(vectors_peak, "SampleID")

metadata_peak <- rownames_to_column(data.frame(data_rare_noinsect_wd@sam_data), "SampleID")
vectors_metadata_peak <- dplyr::inner_join(vectors_peak, metadata_wd, by = "SampleID")

ggplot(vectors_metadata_wd, aes(x = Axis.1, y = Axis.2, color = Year)) + geom_point()

p1 <- plot_ordination(data_rare_noinsect_wd, ordination_pcoa, color="Year") + 
        geom_point(size=5, alpha=0.90) +
        stat_ellipse(aes(fill = Year), geom = "polygon", alpha = .1) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        scale_color_manual(values = c("2021" = "lightcoral", "2023" = "darkred")) +
        scale_fill_manual(values = c("2021" = "lightcoral", "2023" = "darkred"))
        theme(aspect.ratio=1)
png("16S_wd_2021_2023.png", units="in", width=8, height=6, res=300)
p1
dev.off()

p2 <- plot_ordination(data_rare_noinsect_wd, ordination_pcoa, axes = c(1, 3), color="Year") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p2

# Extract sample data
sample_data_df <- as.data.frame(sample_data(data_rare_noinsect_wd))

# Extract site scores from the ordination object
site_scores <- ordination_pcoa$vectors

# Combine scores with sample data
ordination_df <- data.frame(site_scores, sample_data_df)

# Plot the ordination using ggplot2
p <- ggplot(ordination_df, aes_string(x = "Axis.1", y = "Axis.2", color = "Year", shape = "Subplot_Descriptions")) +
  geom_point(size = 4) +
  stat_ellipse(aes(group = Year), type = "norm") + # Add ellipses
  theme_minimal()
p

# Test whether the sampling periods differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_wd)$Year)

# setting up the permutation object
perm1 <- with(sample_data(data_rare_noinsect_wd), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))

perm2 <- with(sample_data(data_rare_noinsect_wd), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_noinsect_wd)$Year, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_wd)$sampling_period, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_wd)$Year*sample_data(data_rare_noinsect_wd)$sampling_period, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_wd)$Year + sample_data(data_rare_noinsect_wd)$sampling_period, by = "margin", permutations = perm1)
```

# WARMED 2021 & 2023
#Subsetting phyloseq object
```{r}
data_rare_noinsect_warm <- subset_samples(data_rare_noinsect_nona, Subplot_Descriptions == "warmed")
summarize_phyloseq(data_rare_noinsect_warm)
```

#Alpha Diversity
```{r}
# order sample data - "sampling_period" and "Subplot_Descriptions"
sample_data(data_rare_noinsect_warm)$sampling_period <- ordered(sample_data(data_rare_noinsect_warm)$sampling_period, c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"))

plot_richness(data_rare_noinsect_warm, x="Year", color="Year", measures=c("Observed"))
plot_richness(data_rare_noinsect_warm, x="Year", measures=c("Observed", "Shannon")) + geom_boxplot()
plot_richness(data_rare_noinsect_warm, x="sampling_period", color="Year", measures=c("Observed"))
plot_richness(data_rare_noinsect_warm, x="sampling_period", measures=c("Observed", "Shannon")) + geom_boxplot()

#png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_warm, x="sampling_period", measures=c("Observed")) + 
        geom_boxplot(aes(fill = sampling_period)) + 
        labs(y = "Observed OTUs", x = "Time of Sampling") +
        facet_grid(.~Year, scale="free") +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
#dev.off()

#png("T7_warmx_16S_shannon_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(data_rare_noinsect_warm, x="sampling_period", measures=c("Shannon")) + 
        geom_boxplot(aes(fill = sampling_period)) + 
        labs(x = "Time of Sampling", y = "Shannon Index") +
        facet_grid(.~Year, scale="free") +
        theme(legend.position = "none", axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
#dev.off()

# analysis of alpha diversity
richness <- estimate_richness(data_rare_noinsect_warm)
head(richness)
kruskal.test(richness$Shannon ~ sample_data(data_rare_noinsect_warm)$Year)
#pairwise.wilcox.test(richness$Shannon, sample_data(data_rare_noinsect_drought_peak)$Year, p.adj = "bonf")
```

#Beta Diversity
```{r}
# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(data_rare_noinsect_warm, method="bray")
ordination_pcoa = ordinate(data_rare_noinsect_warm, method="PCoA", distance=bray_dist)

# https://stackoverflow.com/questions/76465183/plot-ordination-function-of-phyloseq-plots-shapes-double-within-itself
# Save the PCoA vectors data from your ordinated object
vectors_warm <- as.data.frame(ordination_pcoa$vectors)
values_warm <- as.data.frame(ordination_pcoa$values)

library(tibble)
vectors_warm <- rownames_to_column(vectors_warm, "SampleID")

metadata_warm <- rownames_to_column(data.frame(data_rare_noinsect_warm@sam_data), "SampleID")
vectors_metadata_warm <- dplyr::inner_join(vectors_warm, metadata_warm, by = "SampleID")

ggplot(vectors_metadata_wd, aes(x = Axis.1, y = Axis.2, color = Year)) + geom_point()

p1 <- plot_ordination(data_rare_noinsect_warm, ordination_pcoa, color="Year") + 
        geom_point(size=5, alpha=0.90) +
        stat_ellipse(aes(fill = Year), geom = "polygon", alpha = .1) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        scale_color_manual(values = c("2021" = "lightsalmon", "2023" = "red2")) +
        scale_fill_manual(values = c("2021" = "lightsalmon", "2023" = "red2"))
        theme(aspect.ratio=1)
png("16S_warm_2021_2023.png", units="in", width=8, height=6, res=300)
p1
dev.off()

p2 <- plot_ordination(data_rare_noinsect_wd, ordination_pcoa, axes = c(1, 3), color="Year") + 
        geom_point(size=5, alpha=0.80) +
        #stat_ellipse(aes(fill = Drought), geom = "polygon", alpha = .5) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
p2

# Extract sample data
sample_data_df <- as.data.frame(sample_data(data_rare_noinsect_warm))

# Extract site scores from the ordination object
site_scores <- ordination_pcoa$vectors

# Combine scores with sample data
ordination_df <- data.frame(site_scores, sample_data_df)

# Plot the ordination using ggplot2
p <- ggplot(ordination_df, aes_string(x = "Axis.1", y = "Axis.2", color = "Year", shape = "Subplot_Descriptions")) +
  geom_point(size = 4) +
  stat_ellipse(aes(group = Year), type = "norm") + # Add ellipses
  theme_minimal()
p

# Test whether the sampling periods differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_warm)$Year)

# setting up the permutation object
perm1 <- with(sample_data(data_rare_noinsect_warm), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))

perm2 <- with(sample_data(data_rare_noinsect_warm), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_noinsect_warm)$Year, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_warm)$sampling_period, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_noinsect_warm)$Year*sample_data(data_rare_noinsect_warm)$sampling_period, by = "margin", permutations = perm1) # interaction not signigicatn
adonis2(bray_dist ~ sample_data(data_rare_noinsect_warm)$Year + sample_data(data_rare_noinsect_warm)$sampling_period, by = "margin", permutations = perm1)
```


# NMDS
```{r}
ord_16S <- prune_taxa(names(sort(taxa_sums(data_rare_noinsect_drought), TRUE)[1:50]), data_rare_noinsect_drought)

# bray curtis ordination
nmds_bray_16S <- ordinate(
        physeq = ord_16S, 
        method = "NMDS", 
        distance = "bray"
        )

# plot ordination
plot_ordination(nmds_bray_16S, "samples", color = "Subplot_Descriptions")
# Run 20 stress 0.137421
#png("T7_warmx_16S_braycurtis_NMDS_ordination.png", units="in", width=10, height=6, res=300)
plot_ordination(
  physeq = ord_16S,                                                         #phyloseq object
  ordination = nmds_bray_16S) +                                                #ordination
  geom_point(aes(fill = sampling_period, shape = Year), size = 3) + #sets fill color to sampletype
        #stat_ellipse(aes(fill = factor(Year)), geom = "polygon", alpha = .5) +
        theme(legend.title = element_blank())                                      #removes legend title
#dev.off()
```


# 2021 & 2023
# Alpha Diversity
```{r}
#png("T7_warmx_16S_observed_alpha_diversity_no_insects.png", units="in", width=10, height=6, res=300)
plot_richness(rare_16S_noinsect, x="sampling_period", measures=c("Observed")) + 
        geom_boxplot(aes(fill = sampling_period)) + 
        labs(y = "Observed OTUs", x = "Time of Sampling") +
        facet_grid(Subplot_Descriptions~Year, scale="free") +
        theme(legend.position = "none", 
              axis.text.x = element_text(angle = 45, vjust = 1, hjust=1, size = 15),
              axis.text.y = element_text(size = 12),
              axis.title.x = element_text(size=12),
              axis.title.y = element_text(size=12),
              strip.text.x = element_text(size = 12, face = "italic"))
#dev.off()
```

# Beta Diversity
```{r}
# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(rare_16S_noinsect, method="bray")
ordination_pcoa = ordinate(rare_16S_noinsect, method="PCoA", distance=bray_dist)

p1 <- plot_ordination(rare_16S_noinsect, ordination_pcoa, color="Year") + 
        geom_point(size=5, alpha=0.90) +
        stat_ellipse(aes(fill = Year), geom = "polygon", alpha = .1) +
        #labs(x="Axis 1",
        #     y="Axis 2") +
        scale_color_manual(values = c("2021" = "lightsalmon", "2023" = "red2")) +
        scale_fill_manual(values = c("2021" = "lightsalmon", "2023" = "red2"))
        theme(aspect.ratio=1)
p1
```


# no insect samples - order level
```{r}
# rare_16S_noinsect
# class
order <- aggregate_taxa(rare_16S_noinsect, 'Order', verbose = FALSE)

# PCoA plot using bray-curtis as distance
bray_dist = phyloseq::distance(order, method="bray")
ordination = ordinate(order, method="PCoA", distance=bray_dist)

png("T7_warmx_16S_PCOA_order_noinsects_year.png", units="in", width=9, height=6, res=300)
plot_ordination(order, ordination) + 
        geom_point(aes(color = Year), size=4, alpha=0.90) +
        stat_ellipse(aes(color = Year), geom = "polygon", alpha = .1) +
        scale_colour_manual(name = "Year", 
                          breaks = c("2021", "2023"),
                          labels = c("2021", "2023"),
                          values = c("2021" = "#a6bddb", "2023" = "#fb6a4a")) +
        #scale_shape_manual(name = "Sampling Time", 
        #               breaks = c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"),
        #               labels = c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"),
        #               values = c(16, 17, 15, 18)) + 
        theme(aspect.ratio=1)
dev.off()

adonis2(bray_dist ~ sample_data(order)$Subplot_Descriptions + sample_data(order)$sampling_period + sample_data(order)$Year) 

# this works
perm1 <- with(sample_data(order), how(blocks=Rep, plots=Plots(strata=Rep_Footprint, type="free"), within=Within(type="free"), nperm = 999))

# drops footprint - we don't need to include this (based on conversations with Phoebe) - but do we need this when we are looking at multiple years?
perm2 <- with(sample_data(order), how(blocks=Rep, within=Within(type="free"), nperm = 999))

# doesn't work with perm1 - needs a balanced design
adonis2(bray_dist ~ sample_data(order)$Subplot_Descriptions + sample_data(order)$sampling_period, by = "margin", permutations = perm1)

adonis2(bray_dist ~ sample_data(order)$Subplot_Descriptions + sample_data(order)$sampling_period + sample_data(order)$Year, by = "margin", permutations = perm2)

adonis2(bray_dist ~ sample_data(order)$Drought + sample_data(order)$Warming + sample_data(order)$sampling_period, by = "margin", permutations = perm2)

# thoughts - if i have missing samples, for example here with pre drought, maybe make it so all time points have the same amount of samples, but will lose some statistical power by dropping one of my replicates

# betadisper
# betadisper tests whether the dispersion (variability) of samples within groups is similar across groups. If the p-value from the anova test is low (typically <0.05), it suggests that there are significant differences in dispersion between groups. If the dispersion differs significantly between groups, this suggests that group differences in community composition may be driven by more than just differences in group centroids (the average composition of each group). 

# Step 1: Generate Bray-Curtis distance matrix
# generated above:
# bray_dist

# Step 2: Extract the grouping variable from sample_data
group_factor <- sample_data(phylum)$Year  # Ensure 'Subplot_Descriptions' is a column in sample_data

# If it's not a factor, convert it
# group_factor <- as.factor(group_factor)

# Step 3: Run betadisper
betadisper_result <- betadisper(bray_dist, group_factor)

# Step 4: Run ANOVA to check for significant dispersion differences
anova(betadisper_result)
```

# 2021
```{r}
# subset for 2021 data
data_rare_2021 <- subset_samples(data_rare_noinsect, Year%in%c("2021"))
data_rare_2021 <- subset_samples(data_rare_2021, !is.na(sampling_period))
order <- aggregate_taxa(data_rare_2021, 'Order', verbose = FALSE)
data_rare_2021_sample <- sample_data(order)

# PCoA plot using bray-curtis as distance
bray_dist <- phyloseq::distance(data_rare_2021, method="bray", weighted=T)
ordination <- ordinate(data_rare_2021, method="PCoA", distance=bray_dist)

png("16S_pcoa_orderlevel_no_insects.png", units="in", width=8, height=6, res=300)
plot_ordination(data_rare_2021, ordination) + 
        geom_point(aes(color = Subplot_Descriptions, shape = sampling_period), size=4, alpha=0.75) +
        stat_ellipse(aes(color = Subplot_Descriptions), geom = "polygon", alpha = .1) +
            scale_colour_manual(name = "Treatment", 
                          breaks = c("irrigated_control", "ambient", "warmed", "drought", "warmed_drought"),
                          labels = c("Irrigated Control", "Ambient", "Warmed", "Drought", "Warmed + Drought"),
                          values = c("ambient" = "#a6bddb", "drought" = "#687689", "warmed" = "#fb6a4a", "warmed_drought" = "#9D422E")) +
        scale_shape_manual(name = "Sampling Time", 
                       breaks = c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"),
                       labels = c("Pre-Drought", "Peak-Drought", "Post-Drought", "Recovery"),
                       values = c(16, 17, 15, 18))
        #labs(x="Axis 1",
        #     y="Axis 2") +
        theme(aspect.ratio=1)
dev.off()

# Test whether the sampling periods differ significantly from each other using the permutational ANOVA (PERMANOVA) analysis:
set.seed(22)
adonis2(bray_dist ~ sample_data(data_rare_2021)$Subplot_Descriptions)

# setting up the permutation object
perm1 <- with(sample_data(data_rare_2021), how(blocks = Rep_Footprint, plots = Plots(strata=Footprint, type="free"), within=Within(type="free"), nperm = 9999))

perm2 <- with(sample_data(data_rare_2021), how(blocks = Rep_Footprint), within=Within(type="free"), nperm = 9999)

adonis2(bray_dist ~ sample_data(data_rare_2021)$sampling_period, by = "margin", permutations = perm1)
adonis2(bray_dist ~ sample_data(data_rare_2021)$Subplot_Descriptions*sample_data(data_rare_2021)$sampling_period, by = "margin", permutations = perm2) # interaction not significant
adonis2(bray_dist ~ sample_data(data_rare_2021)$Subplot_Descriptions + sample_data(data_rare_2021)$sampling_period, by = "margin", permutations = perm2)
```
